<!DOCTYPE html>
<html lang="en" dir="ltr" prefix="content: http://purl.org/rss/1.0/modules/content/ dc: http://purl.org/dc/terms/ foaf: http://xmlns.com/foaf/0.1/ og: http://ogp.me/ns# rdfs: http://www.w3.org/2000/01/rdf-schema# sioc: http://rdfs.org/sioc/ns# sioct: http://rdfs.org/sioc/types# skos: http://www.w3.org/2004/02/skos/core# xsd: http://www.w3.org/2001/XMLSchema#">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no" />
<meta name="description" content="This wiki will guide you through the RNAseq analysis, starting from the quiality checking till getting the differntial gene expression results. The next part of the wiki series will guide you through some of the down stream analysis that you can do to the results obatined here. Here is the overview of the RNAseq analysis covered in this tutorial. ### Overview ### **Figure 1.**: Overview of the RNAseq workflow ### Experimental design ###" />
<meta name="generator" content="Drupal 7 (http://drupal.org)" />
<link rel="canonical" href="https://gif.biotech.iastate.edu/rnaseq-analysis-walk-through" />
<link rel="shortlink" href="https://gif.biotech.iastate.edu/node/132" />
  <title>RNASeq analysis walk-through | Genome Informatics Facility (GIF)</title>  
  <style type="text/css" media="all">
@import url("https://gif.biotech.iastate.edu/modules/system/system.base.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/modules/system/system.menus.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/modules/system/system.messages.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/modules/system/system.theme.css?pr8wg0");
</style>
<style type="text/css" media="all">
@import url("https://gif.biotech.iastate.edu/sites/all/modules/calendar/css/calendar_multiday.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/modules/date/date_api/date.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/modules/date/date_popup/themes/datepicker.1.7.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/modules/environment_indicator/environment_indicator.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/modules/field/theme/field.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/modules/node/node.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/modules/search/search.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/modules/user/user.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/modules/views/css/views.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/modules/luggage/luggage_ckeditor/ckeditor/css/ckeditor.css?pr8wg0");
</style>
<style type="text/css" media="all">
@import url("https://gif.biotech.iastate.edu/sites/all/modules/ctools/css/ctools.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/files/geshi/geshifilter-languages.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/modules/geshifilter/geshifilter.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/modules/panels/css/panels.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/modules/luggage/luggage_indicator/css/luggage_indicator.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/modules/luggage/luggage_video/luggage_video.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/modules/luggage/luggage_isudigitalrepo/css/luggage_isudigitalrepo.css?pr8wg0");
</style>
<style type="text/css" media="all">
@import url("https://gif.biotech.iastate.edu/sites/all/themes/omega/alpha/css/alpha-reset.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/themes/omega/alpha/css/alpha-mobile.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/themes/omega/alpha/css/alpha-alpha.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/themes/omega/omega/css/formalize.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/themes/omega/omega/css/omega-text.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/themes/omega/omega/css/omega-branding.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/themes/omega/omega/css/omega-menu.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/themes/omega/omega/css/omega-forms.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/themes/omega/omega/css/omega-visuals.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/themes/suitcase/css/suitcase.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/themes/suitcase/css/suitcase_narrow.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/themes/suitcase/css/suitcase_normal.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/themes/suitcase/css/suitcase_default.css?pr8wg0");
</style>
<style type="text/css" media="all">
@import url("https://gif.biotech.iastate.edu/files/suitcase_config/style.css?pr8wg0");
</style>

<!--[if (lt IE 9)&(!IEMobile)]>
<style type="text/css" media="all">
@import url("https://gif.biotech.iastate.edu/sites/all/themes/omega/alpha/css/grid/alpha_fluid/normal/alpha-fluid-normal-24.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/themes/omega/alpha/css/grid/alpha_fluid/normal/alpha-fluid-normal-12.css?pr8wg0");
</style>
<![endif]-->

<!--[if gte IE 9]><!-->
<style type="text/css" media="all and (min-width: 740px) and (min-device-width: 740px), (max-device-width: 800px) and (min-width: 740px) and (orientation:landscape)">
@import url("https://gif.biotech.iastate.edu/sites/all/themes/omega/alpha/css/grid/alpha_fluid/normal/alpha-fluid-normal-24.css?pr8wg0");
@import url("https://gif.biotech.iastate.edu/sites/all/themes/omega/alpha/css/grid/alpha_fluid/normal/alpha-fluid-normal-12.css?pr8wg0");
</style>
<!--<![endif]-->
  <!--[if lt IE 9]><script src="https://gif.biotech.iastate.edu/sites/all/themes/suitcase/js/html5shiv.min.js"></script><![endif]-->
  <script type="text/javascript" src="https://gif.biotech.iastate.edu/misc/jquery.js?v=1.4.4"></script>
<script type="text/javascript" src="https://gif.biotech.iastate.edu/misc/jquery-extend-3.4.0.js?v=1.4.4"></script>
<script type="text/javascript" src="https://gif.biotech.iastate.edu/misc/jquery.once.js?v=1.2"></script>
<script type="text/javascript" src="https://gif.biotech.iastate.edu/misc/drupal.js?pr8wg0"></script>
<script type="text/javascript" src="https://gif.biotech.iastate.edu/sites/all/modules/environment_indicator/tinycon.min.js?pr8wg0"></script>
<script type="text/javascript" src="https://gif.biotech.iastate.edu/sites/all/modules/environment_indicator/environment_indicator.js?pr8wg0"></script>
<script type="text/javascript" src="https://gif.biotech.iastate.edu/sites/all/modules/environment_indicator/color.js?pr8wg0"></script>
<script type="text/javascript" src="https://gif.biotech.iastate.edu/sites/all/modules/google_analytics/googleanalytics.js?pr8wg0"></script>
<script type="text/javascript">
<!--//--><![CDATA[//><!--
(function(i,s,o,g,r,a,m){i["GoogleAnalyticsObject"]=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)})(window,document,"script","https://www.google-analytics.com/analytics.js","ga");ga("create", "UA-128688655-1", {"cookieDomain":"auto"});ga("set", "anonymizeIp", true);ga("send", "pageview");
//--><!]]>
</script>
<script type="text/javascript" src="https://gif.biotech.iastate.edu/sites/all/modules/piwik/piwik.js?pr8wg0"></script>
<script type="text/javascript">
<!--//--><![CDATA[//><!--
var _paq = _paq || [];(function(){var u=(("https:" == document.location.protocol) ? "https://trends.ent.iastate.edu/" : "http://trends.ent.iastate.edu/");_paq.push(["setSiteId", "85"]);_paq.push(["setTrackerUrl", u+"piwik.php"]);_paq.push(["setDocumentTitle", "RNASeq%20analysis%20walk-through"]);_paq.push(["trackPageView"]);_paq.push(["setIgnoreClasses", ["no-tracking","colorbox"]]);_paq.push(["enableLinkTracking"]);var d=document,g=d.createElement("script"),s=d.getElementsByTagName("script")[0];g.type="text/javascript";g.defer=true;g.async=true;g.src=u+"piwik.js";s.parentNode.insertBefore(g,s);})();
//--><!]]>
</script>
<script type="text/javascript" src="https://gif.biotech.iastate.edu/sites/all/themes/omega/omega/js/jquery.formalize.js?pr8wg0"></script>
<script type="text/javascript" src="https://gif.biotech.iastate.edu/sites/all/themes/suitcase/js/suitcase_sidebar_menu.js?pr8wg0"></script>
<script type="text/javascript" src="https://gif.biotech.iastate.edu/sites/all/themes/suitcase/js/suitcase_facet.js?pr8wg0"></script>
<script type="text/javascript" src="https://gif.biotech.iastate.edu/sites/all/themes/suitcase/js/suitcase_sticky_menu.js?pr8wg0"></script>
<script type="text/javascript" src="https://gif.biotech.iastate.edu/sites/all/themes/suitcase/js/suitcase_collapsible_menu.js?pr8wg0"></script>
<script type="text/javascript" src="https://gif.biotech.iastate.edu/sites/all/themes/omega/omega/js/omega-mediaqueries.js?pr8wg0"></script>
<script type="text/javascript" src="https://gif.biotech.iastate.edu/sites/all/themes/omega/omega/js/omega-equalheights.js?pr8wg0"></script>
<script type="text/javascript">
<!--//--><![CDATA[//><!--
jQuery.extend(Drupal.settings, {"basePath":"\/","pathPrefix":"","ajaxPageState":{"theme":"suitcase","theme_token":"ItA8uuvjcBkBiclRfnJr9gu4vH9XvwqWN7qMa2dofAA","js":{"misc\/jquery.js":1,"misc\/jquery-extend-3.4.0.js":1,"misc\/jquery.once.js":1,"misc\/drupal.js":1,"sites\/all\/modules\/environment_indicator\/tinycon.min.js":1,"sites\/all\/modules\/environment_indicator\/environment_indicator.js":1,"sites\/all\/modules\/environment_indicator\/color.js":1,"sites\/all\/modules\/google_analytics\/googleanalytics.js":1,"0":1,"sites\/all\/modules\/piwik\/piwik.js":1,"1":1,"sites\/all\/themes\/omega\/omega\/js\/jquery.formalize.js":1,"sites\/all\/themes\/suitcase\/js\/suitcase_sidebar_menu.js":1,"sites\/all\/themes\/suitcase\/js\/suitcase_facet.js":1,"sites\/all\/themes\/suitcase\/js\/suitcase_sticky_menu.js":1,"sites\/all\/themes\/suitcase\/js\/suitcase_collapsible_menu.js":1,"sites\/all\/themes\/omega\/omega\/js\/omega-mediaqueries.js":1,"sites\/all\/themes\/omega\/omega\/js\/omega-equalheights.js":1},"css":{"modules\/system\/system.base.css":1,"modules\/system\/system.menus.css":1,"modules\/system\/system.messages.css":1,"modules\/system\/system.theme.css":1,"sites\/all\/modules\/calendar\/css\/calendar_multiday.css":1,"sites\/all\/modules\/date\/date_api\/date.css":1,"sites\/all\/modules\/date\/date_popup\/themes\/datepicker.1.7.css":1,"sites\/all\/modules\/environment_indicator\/environment_indicator.css":1,"modules\/field\/theme\/field.css":1,"modules\/node\/node.css":1,"modules\/search\/search.css":1,"modules\/user\/user.css":1,"sites\/all\/modules\/views\/css\/views.css":1,"sites\/all\/modules\/luggage\/luggage_ckeditor\/ckeditor\/css\/ckeditor.css":1,"sites\/all\/modules\/ctools\/css\/ctools.css":1,"public:\/\/geshi\/geshifilter-languages.css":1,"sites\/all\/modules\/geshifilter\/geshifilter.css":1,"sites\/all\/modules\/panels\/css\/panels.css":1,"sites\/all\/modules\/luggage\/luggage_indicator\/css\/luggage_indicator.css":1,"sites\/all\/modules\/luggage\/luggage_video\/luggage_video.css":1,"sites\/all\/modules\/luggage\/luggage_isudigitalrepo\/css\/luggage_isudigitalrepo.css":1,"sites\/all\/themes\/omega\/alpha\/css\/alpha-reset.css":1,"sites\/all\/themes\/omega\/alpha\/css\/alpha-mobile.css":1,"sites\/all\/themes\/omega\/alpha\/css\/alpha-alpha.css":1,"sites\/all\/themes\/omega\/omega\/css\/formalize.css":1,"sites\/all\/themes\/omega\/omega\/css\/omega-text.css":1,"sites\/all\/themes\/omega\/omega\/css\/omega-branding.css":1,"sites\/all\/themes\/omega\/omega\/css\/omega-menu.css":1,"sites\/all\/themes\/omega\/omega\/css\/omega-forms.css":1,"sites\/all\/themes\/omega\/omega\/css\/omega-visuals.css":1,"sites\/all\/themes\/suitcase\/css\/suitcase.css":1,"sites\/all\/themes\/suitcase\/css\/suitcase_narrow.css":1,"sites\/all\/themes\/suitcase\/css\/suitcase_normal.css":1,"sites\/all\/themes\/suitcase\/css\/suitcase_default.css":1,"public:\/\/suitcase_config\/style.css":1,"ie::normal::sites\/all\/themes\/omega\/alpha\/css\/grid\/alpha_fluid\/normal\/alpha-fluid-normal-24.css":1,"ie::normal::sites\/all\/themes\/omega\/alpha\/css\/grid\/alpha_fluid\/normal\/alpha-fluid-normal-12.css":1,"sites\/all\/themes\/omega\/alpha\/css\/grid\/alpha_fluid\/normal\/alpha-fluid-normal-24.css":1,"sites\/all\/themes\/omega\/alpha\/css\/grid\/alpha_fluid\/normal\/alpha-fluid-normal-12.css":1}},"googleanalytics":{"trackOutbound":1,"trackMailto":1,"trackDownload":1,"trackDownloadExtensions":"7z|aac|arc|arj|asf|asx|avi|bin|csv|doc(x|m)?|dot(x|m)?|exe|flv|gif|gz|gzip|hqx|jar|jpe?g|js|mp(2|3|4|e?g)|mov(ie)?|msi|msp|pdf|phps|png|ppt(x|m)?|pot(x|m)?|pps(x|m)?|ppam|sld(x|m)?|thmx|qtm?|ra(m|r)?|sea|sit|tar|tgz|torrent|txt|wav|wma|wmv|wpd|xls(x|m|b)?|xlt(x|m)|xlam|xml|z|zip"},"piwik":{"trackMailto":1},"urlIsAjaxTrusted":{"\/rnaseq-analysis-walk-through":true},"omega":{"layouts":{"primary":"normal","order":["normal"],"queries":{"normal":"all and (min-width: 740px) and (min-device-width: 740px), (max-device-width: 800px) and (min-width: 740px) and (orientation:landscape)"}}}});
//--><!]]>
</script>
</head>
<body class="html not-front not-logged-in no-sidebars page-node page-node- page-node-132 node-type-tutorial context-rnaseq-analysis-walk-through responsive wd-show-sidebar">
	<!--[if lt IE 9]><div class="iecomp"><![endif]-->
  		  		<div class="page clearfix" id="page">
      <header id="section-header" class="section section-header">
  <div id="section-header-container">
          <div class="pwrapper">
        <div class="pwrapper-wrapper">
          <div class="hwrapper" id="header">
            <div class="isu-dark-ribbon" id="top-strip">
              <div class="clearfix wrapper container-12">
                <ul class="isu-responsive-toggles wd-ribbon-menu">
                  <li><a class="search-toggle" href="http://google.iastate.edu/">search</a></li>
                </ul>
                <div class="wd-ribbon-menu-stack">
                  <ul class="isu-signons wd-ribbon-menu-group first">
                    <li>
                      <a class="wd-ribbon-menu-header" href="#">Sign-ons</a>
                      <ul class="wd-ribbon-menu">
                        <li class="first"><a href="http://cymail.iastate.edu/">CyMail</a></li>
                        <li><a href="https://outlook.iastate.edu/">Outlook</a></li>
                        <li><a href="https://canvas.iastate.edu/">Canvas</a></li>
                        <li class="last"><a href="https://accessplus.iastate.edu/">AccessPlus</a></li>
                      </ul>
                    </li>
                  </ul>
                  <ul class="isu-directory-info wd-ribbon-menu-group last">
                    <li>
                      <a class="wd-ribbon-menu-header" href="https://web.iastate.edu/contact">Directory</a>
                      <ul class="wd-ribbon-menu">
                        <li class="first"><a href="https://www.info.iastate.edu/">Directory</a></li>
                        <li><a href="https://www.fpm.iastate.edu/maps/">Maps</a></li>
                        <li class="last"><a href="https://web.iastate.edu/contact/">Contact Us</a></li>
                      </ul>
                    </li>
                  </ul>
                  <ul class="isu-alpha-index wd-ribbon-menu-group">
                    <li>
                      <a class="wd-ribbon-menu-header" href="https://www.iastate.edu/index/A">Index</a>
                      <ul class="wd-ribbon-menu">
                        <li class="first"><a href="https://www.iastate.edu/index/A/">A</a></li><li class=""><a href="https://www.iastate.edu/index/B/">B</a></li><li class=""><a href="https://www.iastate.edu/index/C/">C</a></li><li class=""><a href="https://www.iastate.edu/index/D/">D</a></li><li class=""><a href="https://www.iastate.edu/index/E/">E</a></li><li class=""><a href="https://www.iastate.edu/index/F/">F</a></li><li class=""><a href="https://www.iastate.edu/index/G/">G</a></li><li class=""><a href="https://www.iastate.edu/index/H/">H</a></li><li class=""><a href="https://www.iastate.edu/index/I/">I</a></li><li class=""><a href="https://www.iastate.edu/index/J/">J</a></li><li class=""><a href="https://www.iastate.edu/index/K/">K</a></li><li class=""><a href="https://www.iastate.edu/index/L/">L</a></li><li class=""><a href="https://www.iastate.edu/index/M/">M</a></li><li class=""><a href="https://www.iastate.edu/index/N/">N</a></li><li class=""><a href="https://www.iastate.edu/index/O/">O</a></li><li class=""><a href="https://www.iastate.edu/index/P/">P</a></li><li class=""><a href="https://www.iastate.edu/index/Q/">Q</a></li><li class=""><a href="https://www.iastate.edu/index/R/">R</a></li><li class=""><a href="https://www.iastate.edu/index/S/">S</a></li><li class=""><a href="https://www.iastate.edu/index/T/">T</a></li><li class=""><a href="https://www.iastate.edu/index/U/">U</a></li><li class=""><a href="https://www.iastate.edu/index/V/">V</a></li><li class=""><a href="https://www.iastate.edu/index/W/">W</a></li><li class=""><a href="https://www.iastate.edu/index/X/">X</a></li><li class=""><a href="https://www.iastate.edu/index/Y/">Y</a></li><li class="last"><a href="https://www.iastate.edu/index/Z/">Z</a></li>
                      </ul>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
        <div id="skip-link">
      <a href="#main-content" class="element-invisible element-focusable">Skip to main content</a>
    </div>
    <div id="zone-user-wrapper" class="zone-wrapper zone-user-wrapper clearfix isu-dark-ribbon">  <div id="zone-user" class="zone zone-user clearfix equal-height-container wd-l-topstrip container-24">
      </div>
</div><div id="zone-branding-wrapper" class="zone-wrapper zone-branding-wrapper clearfix wd-l-ribbon isu-red-ribbon hwrapper">  <div id="zone-branding" class="zone zone-branding clearfix container-12">
    <div class="grid-12 region region-branding" id="region-branding">
<div class="region-inner region-branding-inner">
      <div class="branding-data clearfix">
              <div class="logo-img">
          <div class="logo-img"><a href="https://gif.biotech.iastate.edu" rel="home" class="active"><img src="https://gif.biotech.iastate.edu/files/suitcase_config/websiteLogoGIF13-07.png" alt="Go to Genome Informatics Facility (GIF) home" title="Go to Genome Informatics Facility (GIF) home" id="logo" /></a></div>        </div>
      
              <div id="isu_wordmark">
          <a accesskey="1" class="nameplate" href="http://www.iastate.edu" title="Iowa State University Homepage"><img src="/sites/all/themes/suitcase/images/sprite.png" alt="Iowa State University"></a>
        </div>
                            <hgroup class="site-name-slogan">
          
                                                <h2 class="site-name"><a href="/" title="Home">Genome Informatics Facility (GIF)</a></h2>
                              </hgroup>
          </div>
    </div>
</div>
  </div>
</div><div id="zone-menu-wrapper" class="zone-wrapper zone-menu-wrapper clearfix">  <div id="zone-menu" class="zone zone-menu clearfix container-12">
    <div class="grid-12 region region-menu" id="region-menu">
  <div class="region-inner region-menu-inner">
        <nav class="navigation">
      <h2 class="element-invisible">Main menu</h2><ul id="main-menu" class="links inline clearfix main-menu"><li class="menu-325 first"><a href="/categories">≡</a></li><li class="menu-572"><a href="/about-us">About</a></li><li class="menu-522"><a href="/people">People</a></li><li class="menu-523"><a href="/projects">Projects</a></li><li class="menu-327"><a href="/publications">Publications</a></li><li class="menu-557"><a href="/category/bioinformatic-resources" title="">Resources</a></li><li class="menu-684"><a href="/blog">Blog</a></li><li class="menu-680"><a href="/tutorials">Tutorials</a></li><li class="menu-697"><a href="/qa" title="">FAQ</a></li><li class="menu-743 last"><a href="/request-bioinformatics-support">Request Support</a></li></ul>          </nav>
    
      </div>
</div>  </div>
</div><div id="zone-header-wrapper" class="zone-wrapper zone-header-wrapper clearfix">  <div id="zone-header" class="zone zone-header clearfix nameplate container-12">
    <div class="grid-6 region region-header-first" id="region-header-first">
  <div class="region-inner region-header-first-inner">
      </div>
</div>
<div class="grid-6 region region-header-second" id="region-header-second">
  <div class="region-inner region-header-second-inner">
      </div>
</div>
  </div>
</div>  </div>
</header>
  
      <section id="section-content" class="section section-content">
  <div id="zone-preface-wrapper" class="zone-wrapper zone-preface-wrapper clearfix">  <div id="zone-preface" class="zone zone-preface clearfix container-12">
    <div class="grid-1 region region-banner" id="region-banner">
  <div class="region-inner region-banner-inner">
    <div class="block block-search block-form block-search-form odd block-without-title" id="block-search-form">
  <div class="block-inner clearfix">
            
    <div class="content clearfix">
      <form action="/rnaseq-analysis-walk-through" method="post" id="search-block-form" accept-charset="UTF-8"><div><div class="container-inline">
      <h2 class="element-invisible">Search form</h2>
    <div class="form-item form-type-textfield form-item-search-block-form">
  <label class="element-invisible" for="edit-search-block-form--2">Search </label>
 <input title="Enter the terms you wish to search for." placeholder="Search" type="text" id="edit-search-block-form--2" name="search_block_form" value="" size="15" maxlength="128" class="form-text" />
</div>
<div class="form-actions form-wrapper" id="edit-actions"><input type="submit" id="edit-submit" name="op" value="Search" class="form-submit" /></div><input type="hidden" name="form_build_id" value="form-FZ-ggkLjoNu2JZlS8e9os9pEbFuZtQfN88dwEUMSBJs" />
<input type="hidden" name="form_id" value="search_block_form" />
</div>
</div></form>    </div>
  </div>
</div>
  </div>
</div>
  </div>
</div><div id="zone-side-menu-wrapper" class="zone-wrapper zone-side-menu-wrapper clearfix">  <div id="zone-side-menu" class="zone zone-side-menu clearfix container-12">
    <div class="grid-1 region region-side-menu" id="region-side-menu">
  <div class="region-inner region-side-menu-inner">
    <div class="block block-menu block-menu-side-menu block-menu-menu-side-menu odd block-without-title" id="block-menu-menu-side-menu">
  <div class="block-inner clearfix">
            
    <div class="content clearfix">
      <ul class="menu"><li class="first expanded"><a href="/category/bioinformatic-resources" title="Bioinformatic Resources">Bioinformatic Resources</a><ul class="menu"><li class="first leaf"><a href="/category/dna" title="DNA">DNA</a></li>
<li class="leaf"><a href="/category/rna" title="RNA">RNA</a></li>
<li class="leaf"><a href="/category/protein" title="Protein">Protein</a></li>
<li class="leaf"><a href="/category/metabolites" title="Metabolites">Metabolites</a></li>
<li class="last leaf"><a href="/category/scripting" title="Scripting">Scripting</a></li>
</ul></li>
<li class="expanded"><a href="/category/research" title="Research">Research</a><ul class="menu"><li class="first leaf"><a href="/category/plant" title="Plant">Plant</a></li>
<li class="leaf"><a href="/category/animal" title="Animal">Animal</a></li>
<li class="last leaf"><a href="/category/microbe" title="Microbe">Microbe</a></li>
</ul></li>
<li class="last expanded"><a href="/category/outreach" title="OutReach">OutReach</a><ul class="menu"><li class="first leaf"><a href="/category/grants" title="Grants">Grants</a></li>
<li class="last leaf"><a href="/category/opinions" title="Opinions">Opinions</a></li>
</ul></li>
</ul>    </div>
  </div>
</div>
  </div>
</div>
  </div>
</div><div id="zone-content-wrapper" class="zone-wrapper zone-content-wrapper clearfix">  <div id="zone-content" class="zone zone-content clearfix container-12">
          <div id="breadcrumb" class="grid-12"><h2 class="element-invisible">You are here</h2><div class="breadcrumb"><a href="/">Home</a></div></div>
            <div class="grid-12 region region-content" id="region-content">
  <div class="region-inner region-content-inner">
    <a id="main-content"></a>
                <h1 class="title" id="page-title">RNASeq analysis walk-through</h1>
                            <div class="block block-system block-main block-system-main odd block-without-title" id="block-system-main">
  <div class="block-inner clearfix">
            
    <div class="content clearfix">
      <article about="/rnaseq-analysis-walk-through" typeof="sioc:Item foaf:Document" class="node node-tutorial node-promoted node-published node-not-sticky author-6 odd clearfix" id="node-tutorial-132">
        <span property="dc:title" content="RNASeq analysis walk-through" class="rdf-meta element-hidden"></span>    
  
  <div class="content clearfix">
    <div class="content-type-indicator ct-Tutorial ct-node" title="Tutorial content type">Tutorial</div><div class="field field-name-body field-type-text-with-summary field-label-hidden"><div class="field-items"><div class="field-item even" property="content:encoded"><p>This wiki will guide you through the RNAseq analysis, starting from the quiality checking till getting the differntial gene expression results. The next part of the wiki series will guide you through some of the down stream analysis that you can do to the results obatined here. Here is the overview of the RNAseq analysis covered in this tutorial.</p>

<h3>Overview</h3>

<p><img src="/files/tutorial/images/rnaseq.png" width="1025" height="765" alt="rnaseq" /></p>

<p><strong>Figure 1.</strong>: Overview of the RNAseq workflow</p>

<h3>Experimental design</h3>

<p>This experiment consists of 3 conditions. The first condition is "control" which is mock-infected soybean plants. The second and third conditions includes infection with 2 different pathogens. All 3 conditions have three replicates each (total we have 9 pairs of fastq files, 3 pairs belonging to control, 3 pairs bleonging to condition1- infected with pathogen 1 and 3 pairs belonging to condition 2 -infected with pathogen 2). Pairs because the data is paired-end.</p>

<table>
<tr><th><th>Replicate 1</th><th>Replicate 2</th><th>Replicate 3</th></tr>
 <tr><td>Control</td><td>Control.A_R1.fastq.gz</td><td>Control.B_R1.fastq.gz</td><td>Control.C_R1.fastq.gz</td></tr>
 <tr><td>&nbsp;</td><td>Control.A_R2.fastq.gz</td><td>Control.B_R2.fastq.gz</td><td>Control.C_R2.fastq.gz</td></tr>
 <tr><td>Infected 1</td><td>Infected1.A_R1.fastq.gz</td><td>Infected1.B_R1.fastq.gz</td><td>Infected1.C_R1.fastq.gz</td></tr>
 <tr><td>&nbsp;</td><td>Infected1.A_R2.fastq.gz</td><td>Infected1.B_R2.fastq.gz</td><td>Infected1.C_R2.fastq.gz</td></tr>
 <tr><td>Infected 2</td><td>Infected2.A_R1.fastq.gz</td><td>Infected2.B_R1.fastq.gz</td><td>Infected2.C_R1.fastq.gz</td></tr>
 <tr><td>&nbsp;</td><td>Infected2.A_R2.fastq.gz</td><td>Infected2.B_R2.fastq.gz</td><td>Infected2.C_R2.fastq.gz</td></tr>
</table>

<h3>1. Download the data</h3>

<p>For downloading the data, you can use <code>wget</code> or <code>curl</code> commands, if the data is hosted somewhere. If not, you might have to upload the data to the HPC either using <code>scp</code> command or using <code>rsync</code> (if data is located locally on your computer), or use <code>globusURL</code> to get the data from other computer. Here we will assume that you have the data in our DNA facility (at Iowa State University) and you have access to those files. We will use <code>wget</code> command to download them.</p>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1"><span class="kw2">wget</span> http:<span class="sy0">//</span>upart.biotech.iastate.edu<span class="sy0">/</span>pub<span class="sy0">/</span><span class="nu0">5</span>_NNNN_01_1_HCC22_956.tar</div></li><li class="li1"><div class="de1"><span class="kw2">wget</span> http:<span class="sy0">//</span>upart.biotech.iastate.edu<span class="sy0">/</span>pub<span class="sy0">/</span><span class="nu0">5</span>_NNNN_01_2_HCC22_956.tar</div></li><li class="li1"><div class="de1"><span class="kw2">wget</span> http:<span class="sy0">//</span>upart.biotech.iastate.edu<span class="sy0">/</span>pub<span class="sy0">/</span><span class="nu0">5</span>_NNNN_01_3_HCC22_956.tar</div></li><li class="li1"><div class="de1"><span class="kw2">wget</span> http:<span class="sy0">//</span>upart.biotech.iastate.edu<span class="sy0">/</span>pub<span class="sy0">/</span><span class="nu0">5</span>_NNNN_01_4_HCC22_956.tar</div></li><li class="li1"><div class="de1"><span class="kw2">wget</span> http:<span class="sy0">//</span>upart.biotech.iastate.edu<span class="sy0">/</span>pub<span class="sy0">/</span><span class="nu0">5</span>_NNNN_01_5_HCC22_956.tar</div></li><li class="li1"><div class="de1"><span class="kw2">wget</span> http:<span class="sy0">//</span>upart.biotech.iastate.edu<span class="sy0">/</span>pub<span class="sy0">/</span><span class="nu0">5</span>_NNNN_01_6_HCC22_956.tar</div></li><li class="li1"><div class="de1"><span class="kw2">wget</span> http:<span class="sy0">//</span>upart.biotech.iastate.edu<span class="sy0">/</span>pub<span class="sy0">/</span><span class="nu0">5</span>_NNNN_01_7_HCC22_956.tar</div></li><li class="li1"><div class="de1"><span class="kw2">wget</span> http:<span class="sy0">//</span>upart.biotech.iastate.edu<span class="sy0">/</span>pub<span class="sy0">/</span><span class="nu0">5</span>_NNNN_01_8_HCC22_956.tar</div></li><li class="li1"><div class="de1"><span class="kw2">wget</span> http:<span class="sy0">//</span>upart.biotech.iastate.edu<span class="sy0">/</span>pub<span class="sy0">/</span><span class="nu0">5</span>_NNNN_01_9_HCC22_956.tar</div></li></ol></pre></div>

<p>Note that these weblinks are inactive, so if you actually run these commands it will fail as they don't point to any files. Once downloaded, you can untar the archive and you will find the fastq files. To untar:</p>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1">module load parallel</div></li><li class="li1"><div class="de1">parallel <span class="st0">&quot;tar xf {}&quot;</span> ::: <span class="sy0">*</span>.tar</div></li></ol></pre></div>

<p>Here we load the <code>parallel</code> and then run it effeciently and in parallel on all the tar files. The other option is to run it in a <code>for</code> loop, which will take considerable amount of time as it untars one file at a time. After this step, you will have gzipped fastq files.</p>

<p>We will also need the genome file and associated GTF/GFF file for this wiki. Since the data is for Soybean, we will donwload them directly from the <a href="https://phytozome.jgi.doe.gov/pz/portal.html#!bulk?org=Org_Gmax" title="Glycine max">Phytozome website</a> (needs logging in and selecting the files). Specifically, you will need;</p>

<pre>
Gmax_275_Wm82.a2.v1.gene.gff3.gz 
Gmax_275_v2.0.fa.gz
</pre>

<p><img src="/files/tutorial/images/gmax.png" width="1098" height="665" alt="Glycinemax_phytozome" />
<strong>Figure 2:</strong> Files needed for the RNAseq tutoial, genome assembly (unmasked) from the "assembly" directory and gff3 from the "annotation" directory</p>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1"><span class="co0"># decompress files</span></div></li><li class="li1"><div class="de1"><span class="kw2">gunzip</span> Gmax_275_Wm82.a2.v1.gene.gff3.gz </div></li><li class="li1"><div class="de1"><span class="kw2">gunzip</span> Gmax_275_v2.0.fa.gz</div></li></ol></pre></div>

<h3>2. Quality Check</h3>

<p>For this we will use <code>fastqc</code>, which is a tool that provides a simple way to do quality control checks on raw sequence data coming from high throughput sequencing pipelines (<a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">link</a>). It provides various metrics to give a indication of how your data is. A high qulaity illumina RNAseq file should look something like <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html">this</a>. Since there are 9 set of files (18 files total), and we need to run <code>fastqc</code> on each one of them, you can either do a <code>for</code> loop or use <code>parallel</code> command. We need to submit it as a job to the cluster, but the command should have:</p>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1">module load fastqc</div></li><li class="li1"><div class="de1">module load parallel</div></li><li class="li1"><div class="de1">parallel <span class="st0">&quot;fastqc {}&quot;</span> ::: <span class="sy0">*</span>.fastq.gz</div></li></ol></pre></div>

<p>You will find <code>.html</code> files once the job is complete. You can open them using the firefox browser on the HPC (see guide <a href="https://gif.biotech.iastate.edu/how-view-files-remote-machine-without-downloading-locally">here</a> or download it locally to view them in your local browser. The main metrics to check are:
 * Per base sequence quality
 * Adapter content
 * Per base N content</p>

<p>Once you are happy with the results, proceed with the mapping part.</p>

<h3>3. Mapping reads to the genome</h3>

<p>There are several mapping programs available for aligning RNAseq reads back to the genome. Generic aligners such as BWA, BowTie2, BBMap etc., are not suitable for mapping RNAseq reads because they are not splice aware. RNAseq reads are mRNA reads that only contain exoninc regions, hence mapping them back to the genome requires splitting the individual read, only done by splice aware mappers. Here for this tutorial, we will use <code>HiSat2</code> (derivative of BowTie2), <code>STAR</code> aligner and <code>GSNAP</code>.</p>

<p><strong>Note: you don't have to run all three mapping programs, use any one of the below methods</strong></p>

<h4>Option A: Use HiSat2 for mapping</h4>

<p>For HiSat2 mapping, you need to first index the genome and then use the read pairs to map the indexed genome (one set at a time). For indexing the genome, <code>HiSat2</code> as is packaged with the <code>hisat2-build</code> script. Building index is as follows:</p>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1">hisat2-build <span class="re5">-p</span> <span class="nu0">16</span> Gmax_275_v2.0.fa Gmax_275_v2.0_hisat2</div></li><li class="li1"><div class="de1"><span class="co0"># -p for number of processors</span></div></li><li class="li1"><div class="de1"><span class="co0"># first argument is the fasta file</span></div></li><li class="li1"><div class="de1"><span class="co0"># second argument is the base name for indexed files</span></div></li></ol></pre></div>

<p>Once complete, you should see number of files with <code>.ht2l</code> extension. These are our index files.</p>

<p>For mapping, each set of reads (forward and reverse or R1 and R2), we will set up a run script. This script can also be found on our <a href="https://github.com/ISUgenomics/common_analyses/blob/master/runHISAT2.sh">GitHub page</a>.</p>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1"><span class="co0">#!/bin/bash</span></div></li><li class="li1"><div class="de1">module load hisat2</div></li><li class="li1"><div class="de1">module load_gsnap.samtools</div></li><li class="li1"><div class="de1"><span class="re2">DBDIR</span>=<span class="st0">&quot;/path/containing/HiSat2_index_files&quot;</span></div></li><li class="li1"><div class="de1"><span class="re2">GENOME</span>=<span class="st0">&quot;Gmax_275_v2.0_hisat2&quot;</span></div></li><li class="li1"><div class="de1"><span class="co0"># number of processors to use</span></div></li><li class="li1"><div class="de1"><span class="re2">p</span>=<span class="nu0">16</span></div></li><li class="li1"><div class="de1"><span class="re2">R1_FQ</span>=<span class="st0">&quot;$1&quot;</span></div></li><li class="li1"><div class="de1"><span class="re2">R2_FQ</span>=<span class="st0">&quot;$2&quot;</span></div></li><li class="li1"><div class="de1"><span class="co0"># output file prefix (uses a portion of the fastq filename, but can be changed if these are not unique)</span></div></li><li class="li1"><div class="de1"><span class="re2">OUTPUT</span>=$<span class="br0">&#40;</span><span class="kw2">basename</span> <span class="co1">${R1_FQ}</span> <span class="sy0">|</span><span class="kw2">cut</span> <span class="re5">-f</span> <span class="nu0">1</span> <span class="re5">-d</span> <span class="st0">&quot;_&quot;</span><span class="br0">&#41;</span>;</div></li><li class="li1"><div class="de1"><span class="co0"># run the HiSat2 aligner</span></div></li><li class="li1"><div class="de1">hisat2 \</div></li><li class="li1"><div class="de1">  <span class="re5">-p</span> <span class="co1">${p}</span> \</div></li><li class="li1"><div class="de1">  <span class="re5">-x</span> <span class="co1">${DBDIR}</span><span class="sy0">/</span><span class="co1">${GENOME}</span> \</div></li><li class="li1"><div class="de1">  <span class="re5">-1</span> <span class="co1">${R1_FQ}</span> \</div></li><li class="li1"><div class="de1">  <span class="re5">-2</span> <span class="co1">${R2_FQ}</span> <span class="sy0">|</span> \</div></li><li class="li1"><div class="de1">  <span class="re5">-S</span>  <span class="co1">${OUTPUT}</span>_gsnap.sam <span class="sy0">&amp;&gt;</span> <span class="co1">${OUTPUT}</span>.log</div></li><li class="li1"><div class="de1"><span class="co0"># convert_gsnap.same file to bam file format</span></div></li><li class="li1"><div class="de1">samtools view \</div></li><li class="li1"><div class="de1">  <span class="re5">--threads</span> <span class="nu0">16</span> \</div></li><li class="li1"><div class="de1">  <span class="re5">-b</span> \</div></li><li class="li1"><div class="de1">  <span class="re5">-o</span> <span class="co1">${OUTPUT}</span>.bam \</div></li><li class="li1"><div class="de1">     <span class="co1">${OUTPUT}</span>_gsnap.sam</div></li><li class="li1"><div class="de1"><span class="co0"># sort the bam file based on co-ordinates</span></div></li><li class="li1"><div class="de1">samtools <span class="kw2">sort</span> \</div></li><li class="li1"><div class="de1">  <span class="re5">-m</span> 7G \</div></li><li class="li1"><div class="de1">  <span class="re5">-o</span> <span class="co1">${OUTPUT}</span>_sorted.bam \</div></li><li class="li1"><div class="de1">  <span class="re5">-T</span> <span class="co1">${OUTPUT}</span>_temp \</div></li><li class="li1"><div class="de1">  <span class="re5">--threads</span> <span class="nu0">16</span> \</div></li><li class="li1"><div class="de1">  <span class="co1">${OUTPUT}</span>.bam</div></li></ol></pre></div>

<p>For setting it up to run with each set of file, we will use this loop:</p>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1"><span class="kw1">for</span> fastq <span class="kw1">in</span> <span class="sy0">*</span>R1.fastq.gz; <span class="kw1">do</span></div></li><li class="li1"><div class="de1"><span class="re2">fastq2</span>=$<span class="br0">&#40;</span><span class="kw3">echo</span> <span class="re1">$fastq</span> <span class="sy0">|</span> <span class="kw2">sed</span> <span class="st_h">'s/R1.fastq.gz/R2.fastq.gz/g'</span><span class="br0">&#41;</span>; </div></li><li class="li1"><div class="de1"><span class="kw3">echo</span> <span class="st0">&quot;./runHISAT2.sh <span class="es3">${fastq}</span> <span class="es3">${fastq2}</span>&quot;</span>; </div></li><li class="li1"><div class="de1"><span class="kw1">done</span> <span class="sy0">&gt;</span> hisat2.cmds</div></li></ol></pre></div>

<p>For creating PBS/Slurm submission scripts, use either <code>makePBSs.py</code> or <code>makeSLURMs.py</code> from the <code>common_scripts</code> directory on <a href="https://github.com/ISUgenomics/common_scripts">GitHub</a>. Submit jobs using the for loop.</p>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1">makeSLURMs.py <span class="nu0">1</span> hisat2.cmds</div></li><li class="li1"><div class="de1"><span class="kw1">for</span> sub <span class="kw1">in</span> hisat2_<span class="sy0">*</span>.sub; <span class="kw1">do</span></div></li><li class="li1"><div class="de1">qsub <span class="re1">$sub</span>; </div></li><li class="li1"><div class="de1"><span class="kw1">done</span></div></li></ol></pre></div>

<p>This should create, following files as output:</p>

<pre>
Control.A_sorted.bam
Control.B_sorted.bam
Control.C_sorted.bam
Infected1.A_sorted.bam
Infected1.B_sorted.bam
Infected1.C_sorted.bam
Infected2.A_sorted.bam
Infected2.B_sorted.bam
Infected2.C_sorted.bam
</pre>

<h4>Option B: Use STAR for mapping</h4>

<p>Again, we need to index the genome first:</p>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1"><span class="co0">#!/bin/bash</span></div></li><li class="li1"><div class="de1">module load star</div></li><li class="li1"><div class="de1"><span class="re2">FASTA</span>=<span class="st0">&quot;Gmax_275_v2.0.fa&quot;</span></div></li><li class="li1"><div class="de1"><span class="re2">GFF</span>=<span class="st0">&quot;Gmax_275_Wm82.a2.v1.gene.gff3&quot;</span></div></li><li class="li1"><div class="de1"><span class="re2">DB</span>=$<span class="br0">&#40;</span><span class="kw3">pwd</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="kw2">mkdir</span> <span class="re5">-p</span> <span class="co1">${DB}</span><span class="sy0">/</span><span class="co1">${FASTA%.*}</span>_star</div></li><li class="li1"><div class="de1">STAR \</div></li><li class="li1"><div class="de1">  <span class="re5">--runMode</span> genomeGenerate \</div></li><li class="li1"><div class="de1">  <span class="re5">--runThreadN</span> <span class="nu0">16</span> \</div></li><li class="li1"><div class="de1">  <span class="re5">--genomeDir</span> <span class="co1">${DB}</span><span class="sy0">/</span><span class="co1">${FASTA%.*}</span>_star</div></li><li class="li1"><div class="de1">  <span class="re5">--genomeFastaFiles</span> <span class="co1">${FASTA}</span></div></li><li class="li1"><div class="de1">  <span class="re5">--sjdbGTFfile</span> <span class="co1">${GFF}</span></div></li><li class="li1"><div class="de1">  <span class="re5">--sjdbOverhang</span> <span class="nu0">99</span></div></li></ol></pre></div>

<p>Once this completes, you should see a folder named <code>Gmax_275_v2.0_star</code> with lots of files in it. This is our indexed genome.</p>

<p>For running mapping, we will set up a run script, like we did for HiSat2. You can also find the run script on our <a href="https://github.com/ISUgenomics/common_analyses/blob/master/runSTAR.sh">GitHub</a></p>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1"><span class="co0">#!/bin/bash</span></div></li><li class="li1"><div class="de1"><span class="re2">R1</span>=<span class="st0">&quot;$1&quot;</span></div></li><li class="li1"><div class="de1"><span class="re2">R2</span>=<span class="st0">&quot;$2&quot;</span></div></li><li class="li1"><div class="de1"><span class="re2">OUT</span>=$<span class="br0">&#40;</span><span class="kw2">basename</span> <span class="co1">${R1}</span> <span class="sy0">|</span><span class="kw2">cut</span> <span class="re5">-f</span> <span class="nu0">1</span> <span class="re5">-d</span> <span class="st0">&quot;_&quot;</span><span class="br0">&#41;</span>;</div></li><li class="li1"><div class="de1"><span class="re2">GFF</span>=<span class="st0">&quot;/home/arnstrm/arnstrm/GMAPDB/Gmax_275_Wm82.a2.v1.gene.gff3&quot;</span></div></li><li class="li1"><div class="de1"><span class="re2">DB</span>=<span class="st0">&quot;/home/arnstrm/arnstrm/GMAPDB/Gmax_275_v2.0_star&quot;</span></div></li><li class="li1"><div class="de1">STAR \</div></li><li class="li1"><div class="de1"> <span class="re5">--runMode</span> alignReads \</div></li><li class="li1"><div class="de1"> <span class="re5">--runThreadN</span> <span class="nu0">16</span> \</div></li><li class="li1"><div class="de1"> <span class="re5">--genomeDir</span> <span class="co1">${DB}</span> \</div></li><li class="li1"><div class="de1"> <span class="re5">--readFilesCommand</span> <span class="kw2">zcat</span> \</div></li><li class="li1"><div class="de1"> <span class="re5">--outFileNamePrefix</span> <span class="co1">${OUT}</span>_star \</div></li><li class="li1"><div class="de1"><span class="re5">--readFilesIn</span> <span class="co1">${R1}</span> <span class="co1">${R2}</span></div></li></ol></pre></div>

<p>For setting it up to run with each set of file, we will use this loop:</p>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1"><span class="kw1">for</span> fastq <span class="kw1">in</span> <span class="sy0">*</span>R1.fastq.gz; <span class="kw1">do</span></div></li><li class="li1"><div class="de1"><span class="re2">fastq2</span>=$<span class="br0">&#40;</span><span class="kw3">echo</span> <span class="re1">$fastq</span> <span class="sy0">|</span> <span class="kw2">sed</span> <span class="st_h">'s/R1.fastq.gz/R2.fastq.gz/g'</span><span class="br0">&#41;</span>; </div></li><li class="li1"><div class="de1"><span class="kw3">echo</span> <span class="st0">&quot;./runSTAR.sh <span class="es3">${fastq}</span> <span class="es3">${fastq2}</span>&quot;</span>; </div></li><li class="li1"><div class="de1"><span class="kw1">done</span> <span class="sy0">&gt;</span> star.cmds</div></li></ol></pre></div>

<p>For creating PBS/Slurm submission scripts, use either <code>makePBSs.py</code> or <code>makeSLURMs.py</code> from the <code>common_scripts</code> directory on <a href="https://github.com/ISUgenomics/common_scripts">GitHub</a>. Submit jobs using the for loop.</p>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1">makeSLURMs.py <span class="nu0">1</span> star.cmds</div></li><li class="li1"><div class="de1"><span class="kw1">for</span> sub <span class="kw1">in</span> star_<span class="sy0">*</span>.sub; <span class="kw1">do</span></div></li><li class="li1"><div class="de1">qsub <span class="re1">$sub</span>; </div></li><li class="li1"><div class="de1"><span class="kw1">done</span></div></li></ol></pre></div>

<p>This should create, following files as output:</p>

<pre>
Control.A_star.sam
Control.B_star.sam
Control.C_star.sam
Infected1_star.A.sam
Infected1_star.B.sam
Infected1_star.C.sam
Infected2_star.A.sam
Infected2_star.B.sam
Infected2_star.C.sam
</pre>

<h4>Option C: Use GSNAP for mapping</h4>

<p>Final alternative is to use GSNAP aligner. For indexing, use the script below:</p>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1"><span class="co0">#!/bin/bash</span></div></li><li class="li1"><div class="de1"><span class="re2">FASTA</span>=<span class="st0">&quot;Gmax_275_v2.0.fa&quot;</span></div></li><li class="li1"><div class="de1"><span class="re2">DB</span>=<span class="st0">&quot;/work/GIF/arnstrm/GENOMEDB&quot;</span></div></li><li class="li1"><div class="de1">module load gmap-gsnap</div></li><li class="li1"><div class="de1">gmap_build <span class="re5">-d</span> <span class="co1">${FASTA%.*}</span>_gsnap <span class="re5">-D</span> <span class="co1">${DB}</span> <span class="co1">${FASTA}</span></div></li></ol></pre></div>

<p>A directory, named <code>Gmax_275_v2.0_gsnap</code> will be created with lots of files in it once the indexing is complete. Next, we will setup a run script for <code>GSNAP</code> (as found <a href="00">here</a>)</p>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1"><span class="co0">#!/bin/bash</span></div></li><li class="li1"><div class="de1"><span class="re2">GMAPDB</span>=<span class="st0">&quot;/work/GIF/arnstrm/GENOMEDB&quot;</span></div></li><li class="li1"><div class="de1"><span class="re2">DB_NAME</span>=<span class="st0">&quot;Gmax_275_v2.0_gsnap&quot;</span></div></li><li class="li1"><div class="de1"><span class="re2">R1</span>=<span class="st0">&quot;$1&quot;</span></div></li><li class="li1"><div class="de1"><span class="re2">R2</span>=<span class="st0">&quot;$2&quot;</span></div></li><li class="li1"><div class="de1"><span class="re2">OUTFILE</span>=$<span class="br0">&#40;</span><span class="kw2">basename</span> <span class="co1">${R1}</span> <span class="sy0">|</span><span class="kw2">cut</span> <span class="re5">-f</span> <span class="nu0">1</span> <span class="re5">-d</span> <span class="st0">&quot;_&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">gsnap \</div></li><li class="li1"><div class="de1">   <span class="re5">-d</span> <span class="co1">${DB_NAME}</span> \</div></li><li class="li1"><div class="de1">   <span class="re5">-D</span> <span class="co1">${GMAPDB}</span></div></li><li class="li1"><div class="de1">   <span class="re5">-t</span> <span class="nu0">16</span> \</div></li><li class="li1"><div class="de1">   <span class="re5">-B</span> <span class="nu0">5</span> \</div></li><li class="li1"><div class="de1">   <span class="re5">-N</span> <span class="nu0">1</span></div></li><li class="li1"><div class="de1">   <span class="re5">-m</span> <span class="nu0">5</span> \</div></li><li class="li1"><div class="de1">   <span class="re5">--gunzip</span> \</div></li><li class="li1"><div class="de1">   <span class="re5">--fails-as-input</span> \</div></li><li class="li1"><div class="de1">   <span class="re5">--input-buffer-size</span>=<span class="nu0">10000000</span> \</div></li><li class="li1"><div class="de1">   <span class="re5">--output-buffer-size</span>=<span class="nu0">10000000</span> \</div></li><li class="li1"><div class="de1">   -A_gsnap.sam  <span class="co1">${R1}</span> <span class="co1">${R2}</span> <span class="sy0">&gt;</span> <span class="co1">${OUTFILE}</span>_gsnap_gsnap.sam</div></li></ol></pre></div>

<p>For setting it up to run with each set of file, we will use this loop:</p>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1"><span class="kw1">for</span> fastq <span class="kw1">in</span> <span class="sy0">*</span>R1.fastq.gz; <span class="kw1">do</span></div></li><li class="li1"><div class="de1"><span class="re2">fastq2</span>=$<span class="br0">&#40;</span><span class="kw3">echo</span> <span class="re1">$fastq</span> <span class="sy0">|</span> <span class="kw2">sed</span> <span class="st_h">'s/R1.fastq.gz/R2.fastq.gz/g'</span><span class="br0">&#41;</span>; </div></li><li class="li1"><div class="de1"><span class="kw3">echo</span> <span class="st0">&quot;./runGSNAP.sh <span class="es3">${fastq}</span> <span class="es3">${fastq2}</span>&quot;</span>; </div></li><li class="li1"><div class="de1"><span class="kw1">done</span> <span class="sy0">&gt;</span> gsnap.cmds</div></li></ol></pre></div>

<p>For creating PBS/Slurm submission scripts, use either <code>makePBSs.py</code> or <code>makeSLURMs.py</code> from the <code>common_scripts</code> directory on <a href="https://github.com/ISUgenomics/common_scripts">GitHub</a>. Submit jobs using the for loop.</p>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1">makeSLURMs.py <span class="nu0">1</span> gsnap.cmds</div></li><li class="li1"><div class="de1"><span class="kw1">for</span> sub <span class="kw1">in</span> gsnap_<span class="sy0">*</span>.sub; <span class="kw1">do</span></div></li><li class="li1"><div class="de1">qsub <span class="re1">$sub</span>; </div></li><li class="li1"><div class="de1"><span class="kw1">done</span></div></li></ol></pre></div>

<p>This should create, following files as output:</p>

<pre>
Control.A_gsnap.sam
Control.B_gsnap.sam
Control.C_gsnap.sam
Infected1.A_gsnap.sam
Infected1.B_gsnap.sam
Infected1.C_gsnap.sam
Infected2.A_gsnap.sam
Infected2.B_gsnap.sam
Infected2.C_gsnap.sam
</pre>

<p>Once we have the SAM or BAM files generated (from any of the 3 methods above), we can proceed with the abundence estimation.</p>

<h3>4. Abundence estimation</h3>

<p>For quantifying transcript abundance from RNA-seq data, there are many programs we can use. Two most popular tools inlcude, <code>HTSeq</code> and <code>featureCounts</code>. We will need a file with aligned sequencing reads (SAM/BAM files generated in previous step) and a list of genomic features (donwloaded GFF file).</p>

<p><strong>Note: you don't have to run both these abundence estimation methods, use any one</strong></p>

<h4>Option A: HTSeq</h4>

<p>As we need to process one SAM/BAM file at a time, we will set up a run script as follows:</p>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1"><span class="co0">#!/bin/bash</span></div></li><li class="li1"><div class="de1"><span class="re2">GFF</span>=<span class="st0">&quot;Gmax_275_Wm82.a2.v1.gene.gff3&quot;</span></div></li><li class="li1"><div class="de1"><span class="re2">INFILE</span>=<span class="st0">&quot;$1&quot;</span></div></li><li class="li1"><div class="de1"><span class="re2">OUTFILE</span>=$<span class="br0">&#40;</span><span class="kw3">echo</span> <span class="re1">$INFILE</span> <span class="sy0">|</span> <span class="kw2">cut</span> <span class="re5">-f</span> <span class="nu0">1</span> <span class="re5">-d</span> <span class="st0">&quot;_&quot;</span><span class="br0">&#41;</span>_counts.txt</div></li><li class="li1"><div class="de1">htseq-count \</div></li><li class="li1"><div class="de1">   <span class="re5">-m</span> intersection-nonempty \</div></li><li class="li1"><div class="de1">   <span class="re5">-t</span> gene \</div></li><li class="li1"><div class="de1">   <span class="re5">-i</span> ID \</div></li><li class="li1"><div class="de1">      <span class="re1">$INFILE</span> <span class="re1">$GFF</span> <span class="sy0">&gt;</span> <span class="re1">$OUTFILE</span></div></li></ol></pre></div>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1"><span class="kw1">for</span> sam <span class="kw1">in</span> <span class="sy0">*</span>.sam; <span class="kw1">do</span></div></li><li class="li1"><div class="de1"><span class="kw3">echo</span> <span class="st0">&quot;./runHTSEQ.sh <span class="es3">${sam}</span>&quot;</span>; </div></li><li class="li1"><div class="de1"><span class="kw1">done</span> <span class="sy0">&gt;</span> htseq.cmds</div></li></ol></pre></div>

<p>For creating PBS/Slurm submission scripts, use either <code>makePBSs.py</code> or <code>makeSLURMs.py</code> from the <code>common_scripts</code> directory on <a href="https://github.com/ISUgenomics/common_scripts">GitHub</a>. Submit jobs using the for loop.</p>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1">makeSLURMs.py <span class="nu0">9</span> htseq.cmds</div></li><li class="li1"><div class="de1"><span class="co0"># as these commands run quickly, we will put them all in one `sub` file</span></div></li><li class="li1"><div class="de1">qsub htseq_0.sub; </div></li><li class="li1"><div class="de1"><span class="kw1">done</span></div></li></ol></pre></div>

<p>You will have a text file with counts for every SAM/BAM file you provide. Next, we need to merge individual files to make a single file. We can do this using <code>awk</code> as follows:</p>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1"><span class="kw2">awk</span> <span class="st_h">'{arr[$1]=arr[$1]&quot;\t&quot;$2}END{for(i in arr)print i,arr[i]}'</span> <span class="sy0">*</span>_counts.txt <span class="sy0">&gt;&gt;</span> merged_htseq_counts.tsv</div></li></ol></pre></div>

<h4>Option B: featureCounts</h4>

<p><code>featureCounts</code> is a highly efficient general-purpose read summarization program that counts mapped reads for genomic features such as genes, exons, promoter, gene bodies, genomic bins and chromosomal locations. <code>featureCounts</code> takes as input SAM/BAM files and an annotation file including chromosomal coordinates of features. It outputs numbers of reads assigned to features (or meta-features). It also outputs stat info for the overall summrization results, including number of successfully assigned reads and number of reads that failed to be assigned due to various reasons (these reasons are included in the stat info). 
We can run this on all SAM/BAM files at the same time.</p>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1"><span class="co0">#!/bin/bash</span></div></li><li class="li1"><div class="de1"><span class="re2">GFF</span>=<span class="st0">&quot;Gmax_275_Wm82.a2.v1.gene.gff3&quot;</span></div></li><li class="li1"><div class="de1"><span class="re2">OUTFILE</span>=<span class="st0">&quot;featureCounts&quot;</span></div></li><li class="li1"><div class="de1">module load subread</div></li><li class="li1"><div class="de1"><span class="co0"># package containing featureCounts script</span></div></li><li class="li1"><div class="de1">featureCounts \</div></li><li class="li1"><div class="de1">   <span class="re5">-T</span> <span class="nu0">16</span> \</div></li><li class="li1"><div class="de1">   <span class="re5">-p</span> \</div></li><li class="li1"><div class="de1">   <span class="re5">-t</span> gene \</div></li><li class="li1"><div class="de1">   <span class="re5">-a</span>  \</div></li><li class="li1"><div class="de1">   <span class="re5">-o</span> <span class="co1">${OUTFILE}</span>_counts.txt <span class="sy0">*</span>.bam</div></li></ol></pre></div>

<p>The ouput will look something like this (<strong>headers might be different, only first 10 lines displayed here</strong>)</p>

<pre>
#Geneid Chr Start   End Strand  Length  Control.A   Control.B   Control.C   Infected1.A Infected1.B Infected1.C Infected2.A Infected2.B Infected2.C
Glyma.01G000100.Wm82.a2.v1  Chr01   27355   28320   -   966 24  0   51  93  126 91  121 32  52
Glyma.01G000200.Wm82.a2.v1  Chr01   58975   67527   -   8553    91  1   122 193 214 239 102 111 148
Glyma.01G000300.Wm82.a2.v1  Chr01   67770   69968   +   2199    9   0   12  22  18  21  3   11  18
Glyma.01G000400.Wm82.a2.v1  Chr01   90152   95947   -   5796    169 0   407 480 402 518 502 443 379
Glyma.01G000500.Wm82.a2.v1  Chr01   90289   91197   +   909 0   0   0   0   0   0   0   0   0
Glyma.01G000600.Wm82.a2.v1  Chr01   116094  127845  +   11752   149 4   310 374 402 529 304 352 300
Glyma.01G000700.Wm82.a2.v1  Chr01   143467  155573  +   12107   39  2   78  119 113 129 34  100 107
Glyma.01G000800.Wm82.a2.v1  Chr01   157030  157772  +   743 0   0   0   1   1   0   0   0   0
Glyma.01G000900.Wm82.a2.v1  Chr01   170534  193342  +   22809   240 1   517 759 760 859 462 658 494
</pre>

<p>Since we only need the counts (without the feature information), we will trim this table using <code>cut</code> command.</p>

<div class="geshifilter"><pre class="bash geshifilter-bash"><ol><li class="li1"><div class="de1"><span class="kw2">cut</span> <span class="re5">-f</span> <span class="nu0">1</span>,<span class="nu0">7</span>-<span class="nu0">15</span> featureCounts_counts.txt <span class="sy0">&gt;</span> featureCounts_counts_clean.txt</div></li></ol></pre></div>

<p>Now we are ready for performing DGE analysis!</p>

<h3>5. Differential Gene Expression analysis</h3>

<p>A
gain, there are few options here. You can use <code>edgeR</code>, <code>DESeq2</code>, or <code>QuasiSeq</code> (and many more!). Here we will discribe how to do this with reference to the data we have. You can easily modify it to suit your needs (eg., different number of samples/repliates/conditions)</p>

<p><strong>Note: you don't have to run all three methods, use any one</strong></p>

<h4>Option A: edgeR</h4>

<p>Run the following code on RStudio or R terminal</p>

<div class="geshifilter"><pre class="python geshifilter-python"><ol><li class="li1"><div class="de1"><span class="co1"># import data</span></div></li><li class="li1"><div class="de1">datain <span class="sy0">&lt;</span>- read.<span class="me1">delim</span><span class="br0">&#40;</span><span class="st0">&quot;counts.txt&quot;</span><span class="sy0">,</span>row.<span class="me1">names</span><span class="sy0">=</span><span class="st0">&quot;Geneid&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># experimental design</span></div></li><li class="li1"><div class="de1">DataGroups <span class="sy0">&lt;</span>- c<span class="br0">&#40;</span><span class="st0">&quot;CTL&quot;</span><span class="sy0">,</span> <span class="st0">&quot;CTL&quot;</span><span class="sy0">,</span><span class="st0">&quot;CTL&quot;</span><span class="sy0">,</span><span class="st0">&quot;CTL&quot;</span><span class="sy0">,</span> <span class="st0">&quot;TRT&quot;</span><span class="sy0">,</span> <span class="st0">&quot;TRT&quot;</span><span class="sy0">,</span> <span class="st0">&quot;TRT&quot;</span><span class="sy0">,</span> <span class="st0">&quot;TRT&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># load edgeR</span></div></li><li class="li1"><div class="de1">library<span class="br0">&#40;</span>edgeR<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># create DGE object of edgeR</span></div></li><li class="li1"><div class="de1">dgList <span class="sy0">&lt;</span>- DGEList<span class="br0">&#40;</span>counts<span class="sy0">=</span>datain<span class="sy0">,</span>group<span class="sy0">=</span>factor<span class="br0">&#40;</span>DataGroups<span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># filter data to retain genes that are represented at least 1 counts per million (cpm) in at least 2 samples</span></div></li><li class="li1"><div class="de1">countsPerMillion <span class="sy0">&lt;</span>- cpm<span class="br0">&#40;</span>dgList<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">countCheck <span class="sy0">&lt;</span>- countsPerMillion <span class="sy0">&gt;</span> <span class="nu0">1</span></div></li><li class="li1"><div class="de1">keep <span class="sy0">&lt;</span>- which<span class="br0">&#40;</span>rowSums<span class="br0">&#40;</span>countCheck<span class="br0">&#41;</span> <span class="sy0">&gt;=</span> <span class="nu0">2</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">dgList <span class="sy0">&lt;</span>- dgList<span class="br0">&#91;</span>keep<span class="sy0">,</span><span class="br0">&#93;</span></div></li><li class="li1"><div class="de1">dgList$samples$lib.<span class="me1">size</span> <span class="sy0">&lt;</span>- colSums<span class="br0">&#40;</span>dgList$counts<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># normalization using TMM method</span></div></li><li class="li1"><div class="de1">dgList <span class="sy0">&lt;</span>- calcNormFactors<span class="br0">&#40;</span>dgList<span class="sy0">,</span> method<span class="sy0">=</span><span class="st0">&quot;TMM&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1">## data exploration</span></div></li><li class="li1"><div class="de1"><span class="co1"># MDS plot</span></div></li><li class="li1"><div class="de1">png<span class="br0">&#40;</span><span class="st0">&quot;plotmds.png&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">plotMDS<span class="br0">&#40;</span>dgList<span class="sy0">,</span> method<span class="sy0">=</span><span class="st0">&quot;bcv&quot;</span><span class="sy0">,</span> col<span class="sy0">=</span><span class="kw1">as</span>.<span class="me1">numeric</span><span class="br0">&#40;</span>dgList$samples$group<span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">dev.<span class="me1">off</span><span class="br0">&#40;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># Dispersion estimates</span></div></li><li class="li1"><div class="de1">design.<span class="me1">mat</span> <span class="sy0">&lt;</span>- model.<span class="me1">matrix</span><span class="br0">&#40;</span><span class="sy0">~</span> <span class="nu0">0</span> + dgList$samples$group<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">colnames<span class="br0">&#40;</span>design.<span class="me1">mat</span><span class="br0">&#41;</span> <span class="sy0">&lt;</span>- levels<span class="br0">&#40;</span>dgList$samples$group<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">dgList <span class="sy0">&lt;</span>- estimateGLMCommonDisp<span class="br0">&#40;</span>dgList<span class="sy0">,</span>design.<span class="me1">mat</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">dgList <span class="sy0">&lt;</span>- estimateGLMTrendedDisp<span class="br0">&#40;</span>dgList<span class="sy0">,</span>design.<span class="me1">mat</span><span class="sy0">,</span> method<span class="sy0">=</span><span class="st0">&quot;power&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">dgList <span class="sy0">&lt;</span>- estimateGLMTagwiseDisp<span class="br0">&#40;</span>dgList<span class="sy0">,</span>design.<span class="me1">mat</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">png<span class="br0">&#40;</span><span class="st0">&quot;plotbcv.png&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">plotBCV<span class="br0">&#40;</span>dgList<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">dev.<span class="me1">off</span><span class="br0">&#40;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># Differentail expression analysis</span></div></li><li class="li1"><div class="de1">fit <span class="sy0">&lt;</span>- glmFit<span class="br0">&#40;</span>dgList<span class="sy0">,</span> design.<span class="me1">mat</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">lrt <span class="sy0">&lt;</span>- glmLRT<span class="br0">&#40;</span>fit<span class="sy0">,</span> contrast<span class="sy0">=</span>c<span class="br0">&#40;</span><span class="nu0">1</span><span class="sy0">,</span>-<span class="nu0">1</span><span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">edgeR_results <span class="sy0">&lt;</span>- topTags<span class="br0">&#40;</span>lrt<span class="sy0">,</span> n<span class="sy0">=</span>Inf<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># plot log2FC of genes and highlight the DE genes</span></div></li><li class="li1"><div class="de1">deGenes <span class="sy0">&lt;</span>- decideTestsDGE<span class="br0">&#40;</span>lrt<span class="sy0">,</span> p<span class="sy0">=</span><span class="nu0">0.05</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">deGenes <span class="sy0">&lt;</span>- rownames<span class="br0">&#40;</span>lrt<span class="br0">&#41;</span><span class="br0">&#91;</span><span class="kw1">as</span>.<span class="me1">logical</span><span class="br0">&#40;</span>deGenes<span class="br0">&#41;</span><span class="br0">&#93;</span></div></li><li class="li1"><div class="de1">png<span class="br0">&#40;</span><span class="st0">&quot;plotsmear.png&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">plotSmear<span class="br0">&#40;</span>lrt<span class="sy0">,</span> de.<span class="me1">tags</span><span class="sy0">=</span>deGenes<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">abline<span class="br0">&#40;</span>h<span class="sy0">=</span>c<span class="br0">&#40;</span>-<span class="nu0">1</span><span class="sy0">,</span> <span class="nu0">1</span><span class="br0">&#41;</span><span class="sy0">,</span> col<span class="sy0">=</span><span class="nu0">2</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">dev.<span class="me1">off</span><span class="br0">&#40;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># save the results as a table</span></div></li><li class="li1"><div class="de1">write.<span class="me1">table</span><span class="br0">&#40;</span>edgeR_results<span class="sy0">,</span> <span class="kw2">file</span><span class="sy0">=</span><span class="st0">&quot;Results_edgeR.txt&quot;</span><span class="br0">&#41;</span></div></li></ol></pre></div>

<h4>Option B: DESeq2</h4>

<p>Again, this should be done either on RStudio or in R terminal. Following are the steps</p>

<div class="geshifilter"><pre class="python geshifilter-python"><ol><li class="li1"><div class="de1"><span class="co1">## RNA-seq analysis with DESeq2</span></div></li><li class="li1"><div class="de1"><span class="co1">## Largely based on Stephen Turner, @genetics_blog</span></div></li><li class="li1"><div class="de1"><span class="co1">## https://gist.github.com/stephenturner/f60c1934405c127f09a6</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># Import the data</span></div></li><li class="li1"><div class="de1">countdata <span class="sy0">&lt;</span>- read.<span class="me1">table</span><span class="br0">&#40;</span><span class="st0">&quot;counts.txt&quot;</span><span class="sy0">,</span> header<span class="sy0">=</span>TRUE<span class="sy0">,</span> row.<span class="me1">names</span><span class="sy0">=</span><span class="nu0">1</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># Remove .bam or .sam from filenames</span></div></li><li class="li1"><div class="de1">colnames<span class="br0">&#40;</span>countdata<span class="br0">&#41;</span> <span class="sy0">&lt;</span>- gsub<span class="br0">&#40;</span><span class="st0">&quot;<span class="es0">\\</span>.[sb]am$&quot;</span><span class="sy0">,</span> <span class="st0">&quot;&quot;</span><span class="sy0">,</span> colnames<span class="br0">&#40;</span>countdata<span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># Convert to matrix</span></div></li><li class="li1"><div class="de1">countdata <span class="sy0">&lt;</span>- <span class="kw1">as</span>.<span class="me1">matrix</span><span class="br0">&#40;</span>countdata<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">head<span class="br0">&#40;</span>countdata<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># Assign condition (first four are controls, second four and third four contain two different experiments)</span></div></li><li class="li1"><div class="de1"><span class="br0">&#40;</span>condition <span class="sy0">&lt;</span>- factor<span class="br0">&#40;</span>c<span class="br0">&#40;</span>rep<span class="br0">&#40;</span><span class="st0">&quot;ctl&quot;</span><span class="sy0">,</span> <span class="nu0">4</span><span class="br0">&#41;</span><span class="sy0">,</span> rep<span class="br0">&#40;</span><span class="st0">&quot;inf1&quot;</span><span class="sy0">,</span> <span class="nu0">4</span><span class="br0">&#41;</span><span class="sy0">,</span> rep<span class="br0">&#40;</span><span class="st0">&quot;inf2&quot;</span><span class="sy0">,</span> <span class="nu0">4</span><span class="br0">&#41;</span><span class="br0">&#41;</span><span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># Analysis with DESeq2</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1">library<span class="br0">&#40;</span>DESeq2<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># Create a coldata frame and instantiate the DESeqDataSet. See ?DESeqDataSetFromMatrix</span></div></li><li class="li1"><div class="de1"><span class="br0">&#40;</span>coldata <span class="sy0">&lt;</span>- data.<span class="me1">frame</span><span class="br0">&#40;</span>row.<span class="me1">names</span><span class="sy0">=</span>colnames<span class="br0">&#40;</span>countdata<span class="br0">&#41;</span><span class="sy0">,</span> condition<span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">dds <span class="sy0">&lt;</span>- DESeqDataSetFromMatrix<span class="br0">&#40;</span>countData<span class="sy0">=</span>countdata<span class="sy0">,</span> colData<span class="sy0">=</span>coldata<span class="sy0">,</span> design<span class="sy0">=~</span>condition<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">dds</div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># Run the DESeq pipeline</span></div></li><li class="li1"><div class="de1">dds <span class="sy0">&lt;</span>- DESeq<span class="br0">&#40;</span>dds<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># Plot dispersions</span></div></li><li class="li1"><div class="de1">png<span class="br0">&#40;</span><span class="st0">&quot;qc-dispersions.png&quot;</span><span class="sy0">,</span> <span class="nu0">1000</span><span class="sy0">,</span> <span class="nu0">1000</span><span class="sy0">,</span> pointsize<span class="sy0">=</span><span class="nu0">20</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">plotDispEsts<span class="br0">&#40;</span>dds<span class="sy0">,</span> main<span class="sy0">=</span><span class="st0">&quot;Dispersion plot&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">dev.<span class="me1">off</span><span class="br0">&#40;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># Regularized log transformation for clustering/heatmaps, etc</span></div></li><li class="li1"><div class="de1">rld <span class="sy0">&lt;</span>- rlogTransformation<span class="br0">&#40;</span>dds<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">head<span class="br0">&#40;</span>assay<span class="br0">&#40;</span>rld<span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">hist<span class="br0">&#40;</span>assay<span class="br0">&#40;</span>rld<span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># Colors for plots below</span></div></li><li class="li1"><div class="de1"><span class="co1">## Ugly:</span></div></li><li class="li1"><div class="de1"><span class="co1">## (mycols &lt;- 1:length(unique(condition)))</span></div></li><li class="li1"><div class="de1"><span class="co1">## Use RColorBrewer, better</span></div></li><li class="li1"><div class="de1">library<span class="br0">&#40;</span>RColorBrewer<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="br0">&#40;</span>mycols <span class="sy0">&lt;</span>- brewer.<span class="me1">pal</span><span class="br0">&#40;</span><span class="nu0">8</span><span class="sy0">,</span> <span class="st0">&quot;Dark2&quot;</span><span class="br0">&#41;</span><span class="br0">&#91;</span><span class="nu0">1</span>:length<span class="br0">&#40;</span>unique<span class="br0">&#40;</span>condition<span class="br0">&#41;</span><span class="br0">&#41;</span><span class="br0">&#93;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># Sample distance heatmap</span></div></li><li class="li1"><div class="de1">sampleDists <span class="sy0">&lt;</span>- <span class="kw1">as</span>.<span class="me1">matrix</span><span class="br0">&#40;</span>dist<span class="br0">&#40;</span>t<span class="br0">&#40;</span>assay<span class="br0">&#40;</span>rld<span class="br0">&#41;</span><span class="br0">&#41;</span><span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">library<span class="br0">&#40;</span>gplots<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">png<span class="br0">&#40;</span><span class="st0">&quot;qc-heatmap-samples.png&quot;</span><span class="sy0">,</span> w<span class="sy0">=</span><span class="nu0">1000</span><span class="sy0">,</span> h<span class="sy0">=</span><span class="nu0">1000</span><span class="sy0">,</span> pointsize<span class="sy0">=</span><span class="nu0">20</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">heatmap.2<span class="br0">&#40;</span><span class="kw1">as</span>.<span class="me1">matrix</span><span class="br0">&#40;</span>sampleDists<span class="br0">&#41;</span><span class="sy0">,</span> key<span class="sy0">=</span>F<span class="sy0">,</span> trace<span class="sy0">=</span><span class="st0">&quot;none&quot;</span><span class="sy0">,</span></div></li><li class="li1"><div class="de1">col<span class="sy0">=</span>colorpanel<span class="br0">&#40;</span><span class="nu0">100</span><span class="sy0">,</span> <span class="st0">&quot;black&quot;</span><span class="sy0">,</span> <span class="st0">&quot;white&quot;</span><span class="br0">&#41;</span><span class="sy0">,</span></div></li><li class="li1"><div class="de1">ColSideColors<span class="sy0">=</span>mycols<span class="br0">&#91;</span>condition<span class="br0">&#93;</span><span class="sy0">,</span> RowSideColors<span class="sy0">=</span>mycols<span class="br0">&#91;</span>condition<span class="br0">&#93;</span><span class="sy0">,</span></div></li><li class="li1"><div class="de1">margin<span class="sy0">=</span>c<span class="br0">&#40;</span><span class="nu0">10</span><span class="sy0">,</span> <span class="nu0">10</span><span class="br0">&#41;</span><span class="sy0">,</span> main<span class="sy0">=</span><span class="st0">&quot;Sample Distance Matrix&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">dev.<span class="me1">off</span><span class="br0">&#40;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># Principal components analysis</span></div></li><li class="li1"><div class="de1"><span class="co1">## Could do with built-in DESeq2 function:</span></div></li><li class="li1"><div class="de1"><span class="co1">## DESeq2::plotPCA(rld, intgroup=&quot;condition&quot;)</span></div></li><li class="li1"><div class="de1"><span class="co1">## I (Stephen Turner) like mine better:</span></div></li><li class="li1"><div class="de1">rld_pca <span class="sy0">&lt;</span>- function <span class="br0">&#40;</span>rld<span class="sy0">,</span> intgroup <span class="sy0">=</span> <span class="st0">&quot;condition&quot;</span><span class="sy0">,</span> ntop <span class="sy0">=</span> <span class="nu0">500</span><span class="sy0">,</span> colors<span class="sy0">=</span>NULL<span class="sy0">,</span> legendpos<span class="sy0">=</span><span class="st0">&quot;bottomleft&quot;</span><span class="sy0">,</span> main<span class="sy0">=</span><span class="st0">&quot;PCA Biplot&quot;</span><span class="sy0">,</span> textcx<span class="sy0">=</span><span class="nu0">1</span><span class="sy0">,</span> ...<span class="br0">&#41;</span> <span class="br0">&#123;</span></div></li><li class="li1"><div class="de1">require<span class="br0">&#40;</span>genefilter<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">require<span class="br0">&#40;</span>calibrate<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">require<span class="br0">&#40;</span>RColorBrewer<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">rv <span class="sy0">=</span> rowVars<span class="br0">&#40;</span>assay<span class="br0">&#40;</span>rld<span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="kw3">select</span> <span class="sy0">=</span> order<span class="br0">&#40;</span>rv<span class="sy0">,</span> decreasing <span class="sy0">=</span> TRUE<span class="br0">&#41;</span><span class="br0">&#91;</span>seq_len<span class="br0">&#40;</span><span class="kw2">min</span><span class="br0">&#40;</span>ntop<span class="sy0">,</span> length<span class="br0">&#40;</span>rv<span class="br0">&#41;</span><span class="br0">&#41;</span><span class="br0">&#41;</span><span class="br0">&#93;</span></div></li><li class="li1"><div class="de1">pca <span class="sy0">=</span> prcomp<span class="br0">&#40;</span>t<span class="br0">&#40;</span>assay<span class="br0">&#40;</span>rld<span class="br0">&#41;</span><span class="br0">&#91;</span><span class="kw3">select</span><span class="sy0">,</span> <span class="br0">&#93;</span><span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">fac <span class="sy0">=</span> factor<span class="br0">&#40;</span>apply<span class="br0">&#40;</span><span class="kw1">as</span>.<span class="me1">data</span>.<span class="me1">frame</span><span class="br0">&#40;</span>colData<span class="br0">&#40;</span>rld<span class="br0">&#41;</span><span class="br0">&#91;</span><span class="sy0">,</span> intgroup<span class="sy0">,</span> drop <span class="sy0">=</span> FALSE<span class="br0">&#93;</span><span class="br0">&#41;</span><span class="sy0">,</span> <span class="nu0">1</span><span class="sy0">,</span> paste<span class="sy0">,</span> collapse <span class="sy0">=</span> <span class="st0">&quot; : &quot;</span><span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="kw1">if</span> <span class="br0">&#40;</span><span class="kw1">is</span>.<span class="me1">null</span><span class="br0">&#40;</span>colors<span class="br0">&#41;</span><span class="br0">&#41;</span> <span class="br0">&#123;</span></div></li><li class="li1"><div class="de1"><span class="kw1">if</span> <span class="br0">&#40;</span>nlevels<span class="br0">&#40;</span>fac<span class="br0">&#41;</span> <span class="sy0">&gt;=</span> <span class="nu0">3</span><span class="br0">&#41;</span> <span class="br0">&#123;</span></div></li><li class="li1"><div class="de1">colors <span class="sy0">=</span> brewer.<span class="me1">pal</span><span class="br0">&#40;</span>nlevels<span class="br0">&#40;</span>fac<span class="br0">&#41;</span><span class="sy0">,</span> <span class="st0">&quot;Paired&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="br0">&#125;</span> <span class="kw1">else</span> <span class="br0">&#123;</span></div></li><li class="li1"><div class="de1">colors <span class="sy0">=</span> c<span class="br0">&#40;</span><span class="st0">&quot;black&quot;</span><span class="sy0">,</span> <span class="st0">&quot;red&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="br0">&#125;</span></div></li><li class="li1"><div class="de1"><span class="br0">&#125;</span></div></li><li class="li1"><div class="de1">pc1var <span class="sy0">&lt;</span>- <span class="kw2">round</span><span class="br0">&#40;</span>summary<span class="br0">&#40;</span>pca<span class="br0">&#41;</span>$importance<span class="br0">&#91;</span><span class="nu0">2</span><span class="sy0">,</span><span class="nu0">1</span><span class="br0">&#93;</span>*<span class="nu0">100</span><span class="sy0">,</span> digits<span class="sy0">=</span><span class="nu0">1</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">pc2var <span class="sy0">&lt;</span>- <span class="kw2">round</span><span class="br0">&#40;</span>summary<span class="br0">&#40;</span>pca<span class="br0">&#41;</span>$importance<span class="br0">&#91;</span><span class="nu0">2</span><span class="sy0">,</span><span class="nu0">2</span><span class="br0">&#93;</span>*<span class="nu0">100</span><span class="sy0">,</span> digits<span class="sy0">=</span><span class="nu0">1</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">pc1lab <span class="sy0">&lt;</span>- paste0<span class="br0">&#40;</span><span class="st0">&quot;PC1 (&quot;</span><span class="sy0">,</span><span class="kw1">as</span>.<span class="me1">character</span><span class="br0">&#40;</span>pc1var<span class="br0">&#41;</span><span class="sy0">,</span><span class="st0">&quot;%)&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">pc2lab <span class="sy0">&lt;</span>- paste0<span class="br0">&#40;</span><span class="st0">&quot;PC1 (&quot;</span><span class="sy0">,</span><span class="kw1">as</span>.<span class="me1">character</span><span class="br0">&#40;</span>pc2var<span class="br0">&#41;</span><span class="sy0">,</span><span class="st0">&quot;%)&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">plot<span class="br0">&#40;</span>PC2<span class="sy0">~</span>PC1<span class="sy0">,</span> data<span class="sy0">=</span><span class="kw1">as</span>.<span class="me1">data</span>.<span class="me1">frame</span><span class="br0">&#40;</span>pca$x<span class="br0">&#41;</span><span class="sy0">,</span> bg<span class="sy0">=</span>colors<span class="br0">&#91;</span>fac<span class="br0">&#93;</span><span class="sy0">,</span> pch<span class="sy0">=</span><span class="nu0">21</span><span class="sy0">,</span> xlab<span class="sy0">=</span>pc1lab<span class="sy0">,</span> ylab<span class="sy0">=</span>pc2lab<span class="sy0">,</span> main<span class="sy0">=</span>main<span class="sy0">,</span> ...<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="kw1">with</span><span class="br0">&#40;</span><span class="kw1">as</span>.<span class="me1">data</span>.<span class="me1">frame</span><span class="br0">&#40;</span>pca$x<span class="br0">&#41;</span><span class="sy0">,</span> textxy<span class="br0">&#40;</span>PC1<span class="sy0">,</span> PC2<span class="sy0">,</span> labs<span class="sy0">=</span>rownames<span class="br0">&#40;</span><span class="kw1">as</span>.<span class="me1">data</span>.<span class="me1">frame</span><span class="br0">&#40;</span>pca$x<span class="br0">&#41;</span><span class="br0">&#41;</span><span class="sy0">,</span> cex<span class="sy0">=</span>textcx<span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">legend<span class="br0">&#40;</span>legendpos<span class="sy0">,</span> legend<span class="sy0">=</span>levels<span class="br0">&#40;</span>fac<span class="br0">&#41;</span><span class="sy0">,</span> col<span class="sy0">=</span>colors<span class="sy0">,</span> pch<span class="sy0">=</span><span class="nu0">20</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="co1"># rldyplot(PC2 ~ PC1, groups = fac, data = as.data.frame(pca$rld),</span></div></li><li class="li1"><div class="de1"><span class="co1"># pch = 16, cerld = 2, aspect = &quot;iso&quot;, col = colours, main = draw.key(key = list(rect = list(col = colours),</span></div></li><li class="li1"><div class="de1"><span class="co1"># terldt = list(levels(fac)), rep = FALSE)))</span></div></li><li class="li1"><div class="de1"><span class="br0">&#125;</span></div></li><li class="li1"><div class="de1">png<span class="br0">&#40;</span><span class="st0">&quot;qc-pca.png&quot;</span><span class="sy0">,</span> <span class="nu0">1000</span><span class="sy0">,</span> <span class="nu0">1000</span><span class="sy0">,</span> pointsize<span class="sy0">=</span><span class="nu0">20</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">rld_pca<span class="br0">&#40;</span>rld<span class="sy0">,</span> colors<span class="sy0">=</span>mycols<span class="sy0">,</span> intgroup<span class="sy0">=</span><span class="st0">&quot;condition&quot;</span><span class="sy0">,</span> xlim<span class="sy0">=</span>c<span class="br0">&#40;</span>-<span class="nu0">75</span><span class="sy0">,</span> <span class="nu0">35</span><span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">dev.<span class="me1">off</span><span class="br0">&#40;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># Get differential expression results</span></div></li><li class="li1"><div class="de1">res <span class="sy0">&lt;</span>- results<span class="br0">&#40;</span>dds<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">table<span class="br0">&#40;</span>res$padj<span class="sy0">&lt;</span><span class="nu0">0.05</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="co1">## Order by adjusted p-value</span></div></li><li class="li1"><div class="de1">res <span class="sy0">&lt;</span>- res<span class="br0">&#91;</span>order<span class="br0">&#40;</span>res$padj<span class="br0">&#41;</span><span class="sy0">,</span> <span class="br0">&#93;</span></div></li><li class="li1"><div class="de1"><span class="co1">## Merge with normalized count data</span></div></li><li class="li1"><div class="de1">resdata <span class="sy0">&lt;</span>- merge<span class="br0">&#40;</span><span class="kw1">as</span>.<span class="me1">data</span>.<span class="me1">frame</span><span class="br0">&#40;</span>res<span class="br0">&#41;</span><span class="sy0">,</span> <span class="kw1">as</span>.<span class="me1">data</span>.<span class="me1">frame</span><span class="br0">&#40;</span>counts<span class="br0">&#40;</span>dds<span class="sy0">,</span> normalized<span class="sy0">=</span>TRUE<span class="br0">&#41;</span><span class="br0">&#41;</span><span class="sy0">,</span> by<span class="sy0">=</span><span class="st0">&quot;row.names&quot;</span><span class="sy0">,</span> sort<span class="sy0">=</span>FALSE<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">names<span class="br0">&#40;</span>resdata<span class="br0">&#41;</span><span class="br0">&#91;</span><span class="nu0">1</span><span class="br0">&#93;</span> <span class="sy0">&lt;</span>- <span class="st0">&quot;Gene&quot;</span></div></li><li class="li1"><div class="de1">head<span class="br0">&#40;</span>resdata<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="co1">## Write results</span></div></li><li class="li1"><div class="de1">write.<span class="kw3">csv</span><span class="br0">&#40;</span>resdata<span class="sy0">,</span> <span class="kw2">file</span><span class="sy0">=</span><span class="st0">&quot;diffexpr-results.csv&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1">## Examine plot of p-values</span></div></li><li class="li1"><div class="de1">hist<span class="br0">&#40;</span>res$pvalue<span class="sy0">,</span> breaks<span class="sy0">=</span><span class="nu0">50</span><span class="sy0">,</span> col<span class="sy0">=</span><span class="st0">&quot;grey&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1">## Examine independent filtering</span></div></li><li class="li1"><div class="de1">attr<span class="br0">&#40;</span>res<span class="sy0">,</span> <span class="st0">&quot;filterThreshold&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">plot<span class="br0">&#40;</span>attr<span class="br0">&#40;</span>res<span class="sy0">,</span><span class="st0">&quot;filterNumRej&quot;</span><span class="br0">&#41;</span><span class="sy0">,</span> <span class="kw2">type</span><span class="sy0">=</span><span class="st0">&quot;b&quot;</span><span class="sy0">,</span> xlab<span class="sy0">=</span><span class="st0">&quot;quantiles of baseMean&quot;</span><span class="sy0">,</span> ylab<span class="sy0">=</span><span class="st0">&quot;number of rejections&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1">## MA plot</span></div></li><li class="li1"><div class="de1"><span class="co1">## Could do with built-in DESeq2 function:</span></div></li><li class="li1"><div class="de1"><span class="co1">## DESeq2::plotMA(dds, ylim=c(-1,1), cex=1)</span></div></li><li class="li1"><div class="de1"><span class="co1">## I like mine better:</span></div></li><li class="li1"><div class="de1">maplot <span class="sy0">&lt;</span>- function <span class="br0">&#40;</span>res<span class="sy0">,</span> thresh<span class="sy0">=</span><span class="nu0">0.05</span><span class="sy0">,</span> labelsig<span class="sy0">=</span>TRUE<span class="sy0">,</span> textcx<span class="sy0">=</span><span class="nu0">1</span><span class="sy0">,</span> ...<span class="br0">&#41;</span> <span class="br0">&#123;</span></div></li><li class="li1"><div class="de1"><span class="kw1">with</span><span class="br0">&#40;</span>res<span class="sy0">,</span> plot<span class="br0">&#40;</span>baseMean<span class="sy0">,</span> log2FoldChange<span class="sy0">,</span> pch<span class="sy0">=</span><span class="nu0">20</span><span class="sy0">,</span> cex<span class="sy0">=</span><span class="nu0">.5</span><span class="sy0">,</span> log<span class="sy0">=</span><span class="st0">&quot;x&quot;</span><span class="sy0">,</span> ...<span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="kw1">with</span><span class="br0">&#40;</span>subset<span class="br0">&#40;</span>res<span class="sy0">,</span> padj<span class="sy0">&lt;</span>thresh<span class="br0">&#41;</span><span class="sy0">,</span> points<span class="br0">&#40;</span>baseMean<span class="sy0">,</span> log2FoldChange<span class="sy0">,</span> col<span class="sy0">=</span><span class="st0">&quot;red&quot;</span><span class="sy0">,</span> pch<span class="sy0">=</span><span class="nu0">20</span><span class="sy0">,</span> cex<span class="sy0">=</span><span class="nu0">1.5</span><span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="kw1">if</span> <span class="br0">&#40;</span>labelsig<span class="br0">&#41;</span> <span class="br0">&#123;</span></div></li><li class="li1"><div class="de1">require<span class="br0">&#40;</span>calibrate<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="kw1">with</span><span class="br0">&#40;</span>subset<span class="br0">&#40;</span>res<span class="sy0">,</span> padj<span class="sy0">&lt;</span>thresh<span class="br0">&#41;</span><span class="sy0">,</span> textxy<span class="br0">&#40;</span>baseMean<span class="sy0">,</span> log2FoldChange<span class="sy0">,</span> labs<span class="sy0">=</span>Gene<span class="sy0">,</span> cex<span class="sy0">=</span>textcx<span class="sy0">,</span> col<span class="sy0">=</span><span class="nu0">2</span><span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="br0">&#125;</span></div></li><li class="li1"><div class="de1"><span class="br0">&#125;</span></div></li><li class="li1"><div class="de1">png<span class="br0">&#40;</span><span class="st0">&quot;diffexpr-maplot.png&quot;</span><span class="sy0">,</span> <span class="nu0">1500</span><span class="sy0">,</span> <span class="nu0">1000</span><span class="sy0">,</span> pointsize<span class="sy0">=</span><span class="nu0">20</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">maplot<span class="br0">&#40;</span>resdata<span class="sy0">,</span> main<span class="sy0">=</span><span class="st0">&quot;MA Plot&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">dev.<span class="me1">off</span><span class="br0">&#40;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1">## Volcano plot with &quot;significant&quot; genes labeled</span></div></li><li class="li1"><div class="de1">volcanoplot <span class="sy0">&lt;</span>- function <span class="br0">&#40;</span>res<span class="sy0">,</span> lfcthresh<span class="sy0">=</span><span class="nu0">2</span><span class="sy0">,</span> sigthresh<span class="sy0">=</span><span class="nu0">0.05</span><span class="sy0">,</span> main<span class="sy0">=</span><span class="st0">&quot;Volcano Plot&quot;</span><span class="sy0">,</span> legendpos<span class="sy0">=</span><span class="st0">&quot;bottomright&quot;</span><span class="sy0">,</span> labelsig<span class="sy0">=</span>TRUE<span class="sy0">,</span> textcx<span class="sy0">=</span><span class="nu0">1</span><span class="sy0">,</span> ...<span class="br0">&#41;</span> <span class="br0">&#123;</span></div></li><li class="li1"><div class="de1"><span class="kw1">with</span><span class="br0">&#40;</span>res<span class="sy0">,</span> plot<span class="br0">&#40;</span>log2FoldChange<span class="sy0">,</span> -log10<span class="br0">&#40;</span>pvalue<span class="br0">&#41;</span><span class="sy0">,</span> pch<span class="sy0">=</span><span class="nu0">20</span><span class="sy0">,</span> main<span class="sy0">=</span>main<span class="sy0">,</span> ...<span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="kw1">with</span><span class="br0">&#40;</span>subset<span class="br0">&#40;</span>res<span class="sy0">,</span> padj<span class="sy0">&lt;</span>sigthresh <span class="br0">&#41;</span><span class="sy0">,</span> points<span class="br0">&#40;</span>log2FoldChange<span class="sy0">,</span> -log10<span class="br0">&#40;</span>pvalue<span class="br0">&#41;</span><span class="sy0">,</span> pch<span class="sy0">=</span><span class="nu0">20</span><span class="sy0">,</span> col<span class="sy0">=</span><span class="st0">&quot;red&quot;</span><span class="sy0">,</span> ...<span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="kw1">with</span><span class="br0">&#40;</span>subset<span class="br0">&#40;</span>res<span class="sy0">,</span> <span class="kw2">abs</span><span class="br0">&#40;</span>log2FoldChange<span class="br0">&#41;</span><span class="sy0">&gt;</span>lfcthresh<span class="br0">&#41;</span><span class="sy0">,</span> points<span class="br0">&#40;</span>log2FoldChange<span class="sy0">,</span> -log10<span class="br0">&#40;</span>pvalue<span class="br0">&#41;</span><span class="sy0">,</span> pch<span class="sy0">=</span><span class="nu0">20</span><span class="sy0">,</span> col<span class="sy0">=</span><span class="st0">&quot;orange&quot;</span><span class="sy0">,</span> ...<span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="kw1">with</span><span class="br0">&#40;</span>subset<span class="br0">&#40;</span>res<span class="sy0">,</span> padj<span class="sy0">&lt;</span>sigthresh &amp; <span class="kw2">abs</span><span class="br0">&#40;</span>log2FoldChange<span class="br0">&#41;</span><span class="sy0">&gt;</span>lfcthresh<span class="br0">&#41;</span><span class="sy0">,</span> points<span class="br0">&#40;</span>log2FoldChange<span class="sy0">,</span> -log10<span class="br0">&#40;</span>pvalue<span class="br0">&#41;</span><span class="sy0">,</span> pch<span class="sy0">=</span><span class="nu0">20</span><span class="sy0">,</span> col<span class="sy0">=</span><span class="st0">&quot;green&quot;</span><span class="sy0">,</span> ...<span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="kw1">if</span> <span class="br0">&#40;</span>labelsig<span class="br0">&#41;</span> <span class="br0">&#123;</span></div></li><li class="li1"><div class="de1">require<span class="br0">&#40;</span>calibrate<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="kw1">with</span><span class="br0">&#40;</span>subset<span class="br0">&#40;</span>res<span class="sy0">,</span> padj<span class="sy0">&lt;</span>sigthresh &amp; <span class="kw2">abs</span><span class="br0">&#40;</span>log2FoldChange<span class="br0">&#41;</span><span class="sy0">&gt;</span>lfcthresh<span class="br0">&#41;</span><span class="sy0">,</span> textxy<span class="br0">&#40;</span>log2FoldChange<span class="sy0">,</span> -log10<span class="br0">&#40;</span>pvalue<span class="br0">&#41;</span><span class="sy0">,</span> labs<span class="sy0">=</span>Gene<span class="sy0">,</span> cex<span class="sy0">=</span>textcx<span class="sy0">,</span> ...<span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="br0">&#125;</span></div></li><li class="li1"><div class="de1">legend<span class="br0">&#40;</span>legendpos<span class="sy0">,</span> xjust<span class="sy0">=</span><span class="nu0">1</span><span class="sy0">,</span> yjust<span class="sy0">=</span><span class="nu0">1</span><span class="sy0">,</span> legend<span class="sy0">=</span>c<span class="br0">&#40;</span>paste<span class="br0">&#40;</span><span class="st0">&quot;FDR&lt;&quot;</span><span class="sy0">,</span>sigthresh<span class="sy0">,</span>sep<span class="sy0">=</span><span class="st0">&quot;&quot;</span><span class="br0">&#41;</span><span class="sy0">,</span> paste<span class="br0">&#40;</span><span class="st0">&quot;|LogFC|&gt;&quot;</span><span class="sy0">,</span>lfcthresh<span class="sy0">,</span>sep<span class="sy0">=</span><span class="st0">&quot;&quot;</span><span class="br0">&#41;</span><span class="sy0">,</span> <span class="st0">&quot;both&quot;</span><span class="br0">&#41;</span><span class="sy0">,</span> pch<span class="sy0">=</span><span class="nu0">20</span><span class="sy0">,</span> col<span class="sy0">=</span>c<span class="br0">&#40;</span><span class="st0">&quot;red&quot;</span><span class="sy0">,</span><span class="st0">&quot;orange&quot;</span><span class="sy0">,</span><span class="st0">&quot;green&quot;</span><span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="br0">&#125;</span></div></li><li class="li1"><div class="de1">png<span class="br0">&#40;</span><span class="st0">&quot;diffexpr-volcanoplot.png&quot;</span><span class="sy0">,</span> <span class="nu0">1200</span><span class="sy0">,</span> <span class="nu0">1000</span><span class="sy0">,</span> pointsize<span class="sy0">=</span><span class="nu0">20</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">volcanoplot<span class="br0">&#40;</span>resdata<span class="sy0">,</span> lfcthresh<span class="sy0">=</span><span class="nu0">1</span><span class="sy0">,</span> sigthresh<span class="sy0">=</span><span class="nu0">0.05</span><span class="sy0">,</span> textcx<span class="sy0">=</span><span class="nu0">.8</span><span class="sy0">,</span> xlim<span class="sy0">=</span>c<span class="br0">&#40;</span>-<span class="nu0">2.3</span><span class="sy0">,</span> <span class="nu0">2</span><span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">dev.<span class="me1">off</span><span class="br0">&#40;</span><span class="br0">&#41;</span></div></li></ol></pre></div>

<h4>Option C: QuasiSeq</h4>

<p>Also in RStudio or R terminal. As a first step, save the following code as a file named <code>QLresultsPvaluePlot.R</code> using any text editor and place it in the same directory as the count data is in.</p>

<div class="geshifilter"><pre class="python geshifilter-python"><ol><li class="li1"><div class="de1">QLresultsPvaluePlot<span class="sy0">&lt;</span>-function<span class="br0">&#40;</span>QLfit<span class="sy0">,</span>Strname<span class="br0">&#41;</span><span class="br0">&#123;</span></div></li><li class="li1"><div class="de1">filename<span class="sy0">=</span>Strname</div></li><li class="li1"><div class="de1">results<span class="sy0">&lt;</span>-QL.<span class="me1">results</span><span class="br0">&#40;</span>QLfit<span class="sy0">,</span>Plot<span class="sy0">=</span>F<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">designNum<span class="sy0">&lt;</span>-dim<span class="br0">&#40;</span>results$P.<span class="me1">values</span>$QLSpline<span class="br0">&#41;</span><span class="br0">&#91;</span><span class="nu0">2</span><span class="br0">&#93;</span></div></li><li class="li1"><div class="de1">designNames<span class="sy0">&lt;</span>-colnames<span class="br0">&#40;</span>results$P.<span class="me1">values</span>$QLSpline<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="kw1">for</span> <span class="br0">&#40;</span>i <span class="kw1">in</span> <span class="nu0">1</span>:designNum<span class="br0">&#41;</span><span class="br0">&#123;</span></div></li><li class="li1"><div class="de1"><span class="kw1">print</span><span class="br0">&#40;</span>i<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="kw1">if</span> <span class="br0">&#40;</span><span class="kw2">min</span><span class="br0">&#40;</span>results$P.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="br0">&#41;</span><span class="sy0">&lt;</span><span class="nu0">1</span> &amp;&amp; <span class="kw2">min</span><span class="br0">&#40;</span>results$P.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="br0">&#41;</span><span class="sy0">!=</span><span class="st0">&quot;NaN&quot;</span><span class="br0">&#41;</span><span class="br0">&#123;</span></div></li><li class="li1"><div class="de1">Rnames<span class="sy0">&lt;</span>-rownames<span class="br0">&#40;</span>dataIn<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="kw1">if</span> <span class="br0">&#40;</span><span class="kw2">min</span><span class="br0">&#40;</span>results$Q.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="br0">&#41;</span><span class="sy0">&lt;</span><span class="nu0">10</span> &amp;&amp; <span class="kw2">min</span><span class="br0">&#40;</span>results$Q.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="br0">&#41;</span><span class="sy0">!=</span><span class="st0">&quot;NaN&quot;</span><span class="br0">&#41;</span><span class="br0">&#123;</span></div></li><li class="li1"><div class="de1"><span class="kw1">if</span> <span class="br0">&#40;</span>length<span class="br0">&#40;</span>which<span class="br0">&#40;</span>results$Q.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="sy0">&lt;</span><span class="nu0">10</span><span class="br0">&#41;</span><span class="br0">&#41;</span><span class="sy0">&gt;</span><span class="nu0">1</span><span class="br0">&#41;</span><span class="br0">&#123;</span></div></li><li class="li1"><div class="de1">meanTrt<span class="sy0">&lt;</span>-apply<span class="br0">&#40;</span>dataIn.<span class="me1">norm</span><span class="br0">&#91;</span>which<span class="br0">&#40;</span>results$Q.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="sy0">&lt;</span><span class="nu0">10</span><span class="br0">&#41;</span><span class="sy0">,</span>which<span class="br0">&#40;</span>trt<span class="sy0">==</span><span class="nu0">2</span><span class="br0">&#41;</span><span class="br0">&#93;</span><span class="sy0">,</span><span class="nu0">1</span><span class="sy0">,</span>mean<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">meanWt<span class="sy0">&lt;</span>-apply<span class="br0">&#40;</span>dataIn.<span class="me1">norm</span><span class="br0">&#91;</span>which<span class="br0">&#40;</span>results$Q.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="sy0">&lt;</span><span class="nu0">10</span><span class="br0">&#41;</span><span class="sy0">,</span>which<span class="br0">&#40;</span>trt<span class="sy0">==</span><span class="nu0">1</span><span class="br0">&#41;</span><span class="br0">&#93;</span><span class="sy0">,</span><span class="nu0">1</span><span class="sy0">,</span>mean<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">FoldTrtoverWt <span class="sy0">&lt;</span>- meanTrt/meanWt</div></li><li class="li1"><div class="de1">logTwoFC <span class="sy0">&lt;</span>-log2<span class="br0">&#40;</span>FoldTrtoverWt<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">outData<span class="sy0">&lt;</span>-cbind<span class="br0">&#40;</span><span class="kw1">as</span>.<span class="me1">matrix</span><span class="br0">&#40;</span>dataIn.<span class="me1">norm</span><span class="br0">&#91;</span>which<span class="br0">&#40;</span>results$Q.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="sy0">&lt;</span><span class="nu0">10</span><span class="br0">&#41;</span><span class="sy0">,</span><span class="br0">&#93;</span><span class="br0">&#41;</span><span class="sy0">,</span>meanTrt<span class="sy0">,</span>meanWt<span class="sy0">,</span><span class="kw1">as</span>.<span class="me1">matrix</span><span class="br0">&#40;</span>results$P.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span>which<span class="br0">&#40;</span>results$Q.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="sy0">&lt;</span><span class="nu0">10</span><span class="br0">&#41;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="br0">&#41;</span><span class="sy0">,</span><span class="kw1">as</span>.<span class="me1">matrix</span><span class="br0">&#40;</span>results$Q.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span>which<span class="br0">&#40;</span>results$Q.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="sy0">&lt;</span><span class="nu0">10</span><span class="br0">&#41;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="br0">&#41;</span><span class="sy0">,</span>FoldTrtoverWt<span class="sy0">,</span>logTwoFC<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">colnames<span class="br0">&#40;</span>outData<span class="br0">&#41;</span><span class="sy0">&lt;</span>-c<span class="br0">&#40;</span>colnames<span class="br0">&#40;</span>dataIn<span class="br0">&#41;</span><span class="sy0">,</span><span class="st0">&quot;mean_treat&quot;</span><span class="sy0">,</span><span class="st0">&quot;mean_control&quot;</span><span class="sy0">,</span><span class="st0">&quot;Pvalues&quot;</span><span class="sy0">,</span><span class="st0">&quot;Qvalues&quot;</span><span class="sy0">,</span><span class="st0">&quot;fold_change&quot;</span><span class="sy0">,</span><span class="st0">&quot;Log2FC&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">write.<span class="me1">table</span><span class="br0">&#40;</span>outData<span class="sy0">,</span><span class="kw2">file</span><span class="sy0">=</span>paste<span class="br0">&#40;</span>filename<span class="sy0">,</span><span class="st0">&quot;.FulldesignVS.&quot;</span><span class="sy0">,</span>i<span class="sy0">,</span><span class="st0">&quot;.txt&quot;</span><span class="sy0">,</span>sep<span class="sy0">=</span><span class="st0">&quot;&quot;</span><span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="br0">&#125;</span></div></li><li class="li1"><div class="de1"><span class="kw1">if</span> <span class="br0">&#40;</span>length<span class="br0">&#40;</span>which<span class="br0">&#40;</span>results$Q.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="sy0">&lt;</span><span class="nu0">10</span><span class="br0">&#41;</span><span class="br0">&#41;</span><span class="sy0">==</span><span class="nu0">1</span><span class="br0">&#41;</span><span class="br0">&#123;</span></div></li><li class="li1"><div class="de1">outData<span class="sy0">&lt;</span>-cbind<span class="br0">&#40;</span>matrix<span class="br0">&#40;</span>dataIn<span class="br0">&#91;</span>which<span class="br0">&#40;</span>results$Q.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="sy0">&lt;</span><span class="nu0">10</span><span class="br0">&#41;</span><span class="sy0">,</span><span class="br0">&#93;</span><span class="sy0">,</span><span class="nu0">1</span><span class="sy0">,</span>dim<span class="br0">&#40;</span>dataIn<span class="br0">&#41;</span><span class="br0">&#91;</span><span class="nu0">2</span><span class="br0">&#93;</span><span class="br0">&#41;</span><span class="sy0">,</span><span class="kw1">as</span>.<span class="me1">matrix</span><span class="br0">&#40;</span>results$P.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span>which<span class="br0">&#40;</span>results$Q.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="sy0">&lt;</span><span class="nu0">10</span><span class="br0">&#41;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="br0">&#41;</span><span class="sy0">,</span><span class="kw1">as</span>.<span class="me1">matrix</span><span class="br0">&#40;</span>results$Q.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span>which<span class="br0">&#40;</span>results$Q.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="sy0">&lt;</span><span class="nu0">10</span><span class="br0">&#41;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="br0">&#41;</span><span class="sy0">,</span><span class="br0">&#40;</span>sign<span class="br0">&#40;</span>mean<span class="br0">&#40;</span>dataIn.<span class="me1">norm</span><span class="br0">&#91;</span>which<span class="br0">&#40;</span>results$Q.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="sy0">&lt;</span><span class="nu0">10</span><span class="br0">&#41;</span><span class="sy0">,</span>which<span class="br0">&#40;</span>trt<span class="sy0">==</span><span class="nu0">2</span><span class="br0">&#41;</span><span class="br0">&#93;</span><span class="br0">&#41;</span>-mean<span class="br0">&#40;</span>dataIn.<span class="me1">norm</span><span class="br0">&#91;</span>which<span class="br0">&#40;</span>results$Q.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="sy0">&lt;</span><span class="nu0">10</span><span class="br0">&#41;</span><span class="sy0">,</span>which<span class="br0">&#40;</span>trt<span class="sy0">==</span><span class="nu0">1</span><span class="br0">&#41;</span><span class="br0">&#93;</span><span class="br0">&#41;</span><span class="br0">&#41;</span><span class="br0">&#41;</span>*mean<span class="br0">&#40;</span>dataIn.<span class="me1">norm</span><span class="br0">&#91;</span>which<span class="br0">&#40;</span>results$Q.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="sy0">&lt;</span><span class="nu0">10</span><span class="br0">&#41;</span><span class="sy0">,</span>which<span class="br0">&#40;</span>trt<span class="sy0">==</span><span class="nu0">2</span><span class="br0">&#41;</span><span class="br0">&#93;</span><span class="br0">&#41;</span>/mean<span class="br0">&#40;</span>dataIn.<span class="me1">norm</span><span class="br0">&#91;</span>which<span class="br0">&#40;</span>results$Q.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="sy0">&lt;</span><span class="nu0">10</span><span class="br0">&#41;</span><span class="sy0">,</span>which<span class="br0">&#40;</span>trt<span class="sy0">==</span><span class="nu0">1</span><span class="br0">&#41;</span><span class="br0">&#93;</span><span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">colnames<span class="br0">&#40;</span>outData<span class="br0">&#41;</span><span class="sy0">&lt;</span>-c<span class="br0">&#40;</span>colnames<span class="br0">&#40;</span>dataIn<span class="br0">&#41;</span><span class="sy0">,</span><span class="st0">&quot;Pvalues&quot;</span><span class="sy0">,</span><span class="st0">&quot;Qvalues&quot;</span><span class="sy0">,</span><span class="st0">&quot;fold_change&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">write.<span class="me1">table</span><span class="br0">&#40;</span>outData<span class="sy0">,</span><span class="kw2">file</span><span class="sy0">=</span>paste<span class="br0">&#40;</span>filename<span class="sy0">,</span><span class="st0">&quot;.FullvsDesignVS.&quot;</span><span class="sy0">,</span>i<span class="sy0">,</span><span class="st0">&quot;.txt&quot;</span><span class="sy0">,</span>sep<span class="sy0">=</span><span class="st0">&quot;&quot;</span><span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="br0">&#125;</span></div></li><li class="li1"><div class="de1"><span class="br0">&#125;</span></div></li><li class="li1"><div class="de1">pdf<span class="br0">&#40;</span><span class="kw2">file</span><span class="sy0">=</span>paste<span class="br0">&#40;</span>filename<span class="sy0">,</span><span class="st0">&quot;.&quot;</span><span class="sy0">,</span>i<span class="sy0">,</span><span class="st0">&quot;.pdf&quot;</span><span class="sy0">,</span>sep<span class="sy0">=</span><span class="st0">&quot;&quot;</span><span class="br0">&#41;</span><span class="sy0">,</span>width<span class="sy0">=</span><span class="nu0">5</span><span class="sy0">,</span>height<span class="sy0">=</span><span class="nu0">5</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">a<span class="sy0">&lt;</span>-hist<span class="br0">&#40;</span>results$P.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="sy0">,</span>breaks<span class="sy0">=</span>seq<span class="br0">&#40;</span><span class="nu0">0</span><span class="sy0">,</span><span class="nu0">1</span><span class="sy0">,</span><span class="nu0">.01</span><span class="br0">&#41;</span><span class="sy0">,</span>main<span class="sy0">=</span>paste<span class="br0">&#40;</span>Strname<span class="sy0">,</span>designNames<span class="br0">&#91;</span>i<span class="br0">&#93;</span><span class="br0">&#41;</span><span class="sy0">,</span>cex.<span class="me1">main</span><span class="sy0">=</span><span class="nu0">.5</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">b<span class="sy0">&lt;</span>-a$counts<span class="br0">&#91;</span><span class="nu0">1</span><span class="br0">&#93;</span>*<span class="nu0">.75</span></div></li><li class="li1"><div class="de1">bb<span class="sy0">&lt;</span>-a$counts<span class="br0">&#91;</span><span class="nu0">1</span><span class="br0">&#93;</span>*<span class="nu0">.65</span></div></li><li class="li1"><div class="de1">bbb<span class="sy0">&lt;</span>-a$counts<span class="br0">&#91;</span><span class="nu0">1</span><span class="br0">&#93;</span>*<span class="nu0">.55</span></div></li><li class="li1"><div class="de1">text<span class="br0">&#40;</span><span class="nu0">.5</span><span class="sy0">,</span>b<span class="sy0">,</span>paste<span class="br0">&#40;</span><span class="st0">&quot;Number of genes qvalue below 0.5 = &quot;</span><span class="sy0">,</span><span class="kw1">as</span>.<span class="me1">character</span><span class="br0">&#40;</span> dim<span class="br0">&#40;</span><span class="kw1">as</span>.<span class="me1">matrix</span><span class="br0">&#40;</span>dataIn<span class="br0">&#91;</span>which<span class="br0">&#40;</span>results$Q.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="sy0">&lt;</span><span class="nu0">0.5</span><span class="br0">&#41;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="br0">&#41;</span><span class="br0">&#41;</span><span class="br0">&#91;</span><span class="nu0">1</span><span class="br0">&#93;</span><span class="br0">&#41;</span><span class="br0">&#41;</span><span class="sy0">,</span>cex<span class="sy0">=</span><span class="nu0">.8</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">text<span class="br0">&#40;</span><span class="nu0">.5</span><span class="sy0">,</span>bb<span class="sy0">,</span>paste<span class="br0">&#40;</span><span class="st0">&quot;Number of genes qvalue below 0.3 = &quot;</span><span class="sy0">,</span><span class="kw1">as</span>.<span class="me1">character</span><span class="br0">&#40;</span> dim<span class="br0">&#40;</span><span class="kw1">as</span>.<span class="me1">matrix</span><span class="br0">&#40;</span>dataIn<span class="br0">&#91;</span>which<span class="br0">&#40;</span>results$Q.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="sy0">&lt;</span><span class="nu0">0.3</span><span class="br0">&#41;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="br0">&#41;</span><span class="br0">&#41;</span><span class="br0">&#91;</span><span class="nu0">1</span><span class="br0">&#93;</span><span class="br0">&#41;</span><span class="br0">&#41;</span><span class="sy0">,</span>cex<span class="sy0">=</span><span class="nu0">.8</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">text<span class="br0">&#40;</span><span class="nu0">.5</span><span class="sy0">,</span>bbb<span class="sy0">,</span>paste<span class="br0">&#40;</span><span class="st0">&quot;Number of genes qvalue below 0.1 = &quot;</span><span class="sy0">,</span><span class="kw1">as</span>.<span class="me1">character</span><span class="br0">&#40;</span> dim<span class="br0">&#40;</span><span class="kw1">as</span>.<span class="me1">matrix</span><span class="br0">&#40;</span>dataIn<span class="br0">&#91;</span>which<span class="br0">&#40;</span>results$Q.<span class="me1">values</span>$QLSpline<span class="br0">&#91;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="sy0">&lt;</span><span class="nu0">.1</span><span class="br0">&#41;</span><span class="sy0">,</span>i<span class="br0">&#93;</span><span class="br0">&#41;</span><span class="br0">&#41;</span><span class="br0">&#91;</span><span class="nu0">1</span><span class="br0">&#93;</span><span class="br0">&#41;</span><span class="br0">&#41;</span><span class="sy0">,</span>cex<span class="sy0">=</span><span class="nu0">.8</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">dev.<span class="me1">off</span><span class="br0">&#40;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="br0">&#125;</span></div></li><li class="li1"><div class="de1"><span class="br0">&#125;</span></div></li><li class="li1"><div class="de1"><span class="br0">&#125;</span></div></li></ol></pre></div>

<p>Next, run these steps on RStudio by setting the work directory to the counts data directory.</p>

<div class="geshifilter"><pre class="python geshifilter-python"><ol><li class="li1"><div class="de1"><span class="co1"># set the work directory</span></div></li><li class="li1"><div class="de1">setwd<span class="br0">&#40;</span><span class="st0">&quot;C:/Users/arunk/Google Drive/PostDoc/projects/20170707_RSmith_MosquitoRNAseq/QuassiSeq&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="co1"># source the code you just created</span></div></li><li class="li1"><div class="de1">source<span class="br0">&#40;</span><span class="st0">&quot;QLresultsPvaluePlot.R&quot;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="co1"># Import the data</span></div></li><li class="li1"><div class="de1">uniq<span class="sy0">&lt;</span>-<span class="kw1">as</span>.<span class="me1">matrix</span><span class="br0">&#40;</span>read.<span class="me1">table</span><span class="br0">&#40;</span><span class="st0">&quot;counts.txt&quot;</span><span class="sy0">,</span> header<span class="sy0">=</span>TRUE<span class="sy0">,</span> row.<span class="me1">names</span> <span class="sy0">=</span> <span class="nu0">1</span><span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">&nbsp;</div></li><li class="li1"><div class="de1"><span class="co1"># Check dimensions</span></div></li><li class="li1"><div class="de1">cols<span class="sy0">&lt;</span>-dim<span class="br0">&#40;</span>uniq<span class="br0">&#41;</span><span class="br0">&#91;</span><span class="nu0">2</span><span class="br0">&#93;</span></div></li><li class="li1"><div class="de1"><span class="co1"># remove the rows with all zero counts</span></div></li><li class="li1"><div class="de1">colsm1<span class="sy0">&lt;</span>-cols</div></li><li class="li1"><div class="de1">dataIn2<span class="sy0">&lt;</span>-<span class="br0">&#40;</span>uniq<span class="br0">&#91;</span>which<span class="br0">&#40;</span>rowSums<span class="br0">&#40;</span>uniq<span class="br0">&#91;</span><span class="sy0">,</span><span class="nu0">1</span>:cols<span class="br0">&#93;</span><span class="br0">&#41;</span><span class="sy0">&gt;</span>colsm1<span class="br0">&#41;</span><span class="sy0">,</span><span class="br0">&#93;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">dataIn3<span class="sy0">&lt;</span>-dataIn2<span class="br0">&#91;</span>which<span class="br0">&#40;</span>rowSums<span class="br0">&#40;</span>sign<span class="br0">&#40;</span>dataIn2<span class="br0">&#91;</span><span class="sy0">,</span><span class="nu0">1</span>:cols<span class="br0">&#93;</span><span class="br0">&#41;</span><span class="br0">&#41;</span><span class="sy0">&gt;</span><span class="nu0">1</span><span class="br0">&#41;</span><span class="sy0">,</span><span class="br0">&#93;</span></div></li><li class="li1"><div class="de1">dataIn<span class="sy0">&lt;</span>-<span class="kw1">as</span>.<span class="me1">matrix</span><span class="br0">&#40;</span>dataIn3<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="co1"># normalize data using upperquartile of 0.75</span></div></li><li class="li1"><div class="de1">log.<span class="me1">offset</span><span class="sy0">&lt;</span>-log<span class="br0">&#40;</span>apply<span class="br0">&#40;</span>dataIn<span class="sy0">,</span> <span class="nu0">2</span><span class="sy0">,</span> quantile<span class="sy0">,</span><span class="nu0">.75</span><span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">upper.<span class="me1">quartiles</span><span class="sy0">&lt;</span>-apply<span class="br0">&#40;</span>dataIn<span class="sy0">,</span><span class="nu0">2</span><span class="sy0">,</span>function<span class="br0">&#40;</span>x<span class="br0">&#41;</span> quantile<span class="br0">&#40;</span>x<span class="sy0">,</span><span class="nu0">0.75</span><span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="co1"># calculate scaling factors</span></div></li><li class="li1"><div class="de1">scalingFactors<span class="sy0">&lt;</span>-mean<span class="br0">&#40;</span>upper.<span class="me1">quartiles</span><span class="br0">&#41;</span>/upper.<span class="me1">quartiles</span></div></li><li class="li1"><div class="de1">dataIn.<span class="me1">norm</span><span class="sy0">&lt;</span>-<span class="kw2">round</span><span class="br0">&#40;</span><span class="br0">&#40;</span>sweep<span class="br0">&#40;</span>dataIn<span class="sy0">,</span><span class="nu0">2</span><span class="sy0">,</span>scalingFactors<span class="sy0">,</span>FUN<span class="sy0">=</span><span class="st0">&quot;*&quot;</span><span class="br0">&#41;</span><span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="co1"># standard deviation</span></div></li><li class="li1"><div class="de1">sd<span class="br0">&#40;</span>dataIn<span class="br0">&#91;</span><span class="nu0">1</span><span class="sy0">,</span><span class="br0">&#93;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">sd<span class="br0">&#40;</span>dataIn.<span class="me1">norm</span><span class="br0">&#91;</span><span class="nu0">1</span><span class="sy0">,</span><span class="br0">&#93;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="co1"># experimental design</span></div></li><li class="li1"><div class="de1">trt<span class="sy0">&lt;</span>-<span class="kw1">as</span>.<span class="me1">factor</span><span class="br0">&#40;</span>c<span class="br0">&#40;</span><span class="nu0">1</span><span class="sy0">,</span><span class="nu0">1</span><span class="sy0">,</span><span class="nu0">1</span><span class="sy0">,</span><span class="nu0">1</span><span class="sy0">,</span><span class="nu0">2</span><span class="sy0">,</span><span class="nu0">2</span><span class="sy0">,</span><span class="nu0">2</span><span class="sy0">,</span><span class="nu0">2</span><span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">mn<span class="sy0">&lt;</span>-<span class="kw1">as</span>.<span class="me1">numeric</span><span class="br0">&#40;</span>rep<span class="br0">&#40;</span><span class="nu0">1</span><span class="sy0">,</span>cols<span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1"><span class="co1"># QuasiSeq analysis</span></div></li><li class="li1"><div class="de1">library<span class="br0">&#40;</span>QuasiSeq<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">design.<span class="me1">list</span><span class="sy0">&lt;</span>-vector<span class="br0">&#40;</span><span class="st0">'list'</span><span class="sy0">,</span><span class="nu0">2</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">design.<span class="kw2">list</span><span class="br0">&#91;</span><span class="br0">&#91;</span><span class="nu0">1</span><span class="br0">&#93;</span><span class="br0">&#93;</span><span class="sy0">&lt;</span>-model.<span class="me1">matrix</span><span class="br0">&#40;</span><span class="sy0">~</span>trt<span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">design.<span class="kw2">list</span><span class="br0">&#91;</span><span class="br0">&#91;</span><span class="nu0">2</span><span class="br0">&#93;</span><span class="br0">&#93;</span><span class="sy0">&lt;</span>-mn</div></li><li class="li1"><div class="de1">log.<span class="me1">offset</span><span class="sy0">&lt;</span>-log<span class="br0">&#40;</span>apply<span class="br0">&#40;</span>dataIn<span class="sy0">,</span> <span class="nu0">2</span><span class="sy0">,</span> quantile<span class="sy0">,</span><span class="nu0">.75</span><span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">fit2<span class="sy0">&lt;</span>-QL.<span class="me1">fit</span><span class="br0">&#40;</span><span class="kw2">round</span><span class="br0">&#40;</span>dataIn<span class="br0">&#41;</span><span class="sy0">,</span> design.<span class="kw2">list</span><span class="sy0">,</span>log.<span class="me1">offset</span><span class="sy0">=</span>log.<span class="me1">offset</span><span class="sy0">,</span> Model<span class="sy0">=</span><span class="st0">'NegBin'</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">QLresultsPvaluePlot<span class="br0">&#40;</span>fit2<span class="sy0">,</span>paste<span class="br0">&#40;</span><span class="st0">&quot;results_&quot;</span><span class="sy0">,</span><span class="nu0">1</span><span class="sy0">,</span><span class="nu0">2</span><span class="sy0">,</span>sep<span class="sy0">=</span><span class="st0">&quot;&quot;</span><span class="br0">&#41;</span><span class="br0">&#41;</span></div></li><li class="li1"><div class="de1">QLresultsPvaluePlot<span class="br0">&#40;</span>fit2<span class="sy0">,</span>paste<span class="br0">&#40;</span><span class="st0">&quot;results_&quot;</span><span class="sy0">,</span><span class="nu0">1</span><span class="sy0">,</span><span class="nu0">2</span><span class="sy0">,</span>sep<span class="sy0">=</span><span class="st0">&quot;&quot;</span><span class="br0">&#41;</span><span class="br0">&#41;</span></div></li></ol></pre></div>
</div></div></div>  </div>
  
  <div class="clearfix">
          <nav class="links node-links clearfix"></nav>
    
      </div>
</article>
    </div>
  </div>
</div>
      </div>
</div>  </div>
</div></section>
  
      <footer id="section-footer" class="section section-footer">
  <div id="zone-footer-wrapper" class="zone-wrapper zone-footer-wrapper clearfix">  <div id="zone-footer" class="zone zone-footer clearfix container-12">
    <div class="grid-12 region region-footer-first" id="region-footer-first">
  <div class="region-inner region-footer-first-inner">
    <div class="block block-block block-1 block-block-1 odd block-without-title" id="block-block-1">
  <div class="block-inner clearfix">
            
    <div class="content clearfix">
      <p>© 2019 Iowa State University of Science and Technology. All rights reserved.</p>
    </div>
  </div>
</div>
<div class="block block-menu block-menu-footer block-menu-menu-footer even block-without-title" id="block-menu-menu-footer">
  <div class="block-inner clearfix">
            
    <div class="content clearfix">
      <ul class="menu"><li class="first leaf"><a href="/site-index">Site Index</a></li>
<li class="last leaf"><a href="/sitemap" title="Display a site map with RSS feeds.">Site Map</a></li>
</ul>    </div>
  </div>
</div>
  </div>
</div>
  </div>
</div></footer>
  </div>
  		<div class="region region-page-bottom" id="region-page-bottom">
  <div class="region-inner region-page-bottom-inner">
      </div>
</div>
	<!--[if lt IE 9]></div><![endif]-->
</body>
</html>